<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>IntegratingDataBioc • IntegratingDataBioc</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="IntegratingDataBioc">
<meta property="og:description" content="Researchers use ChIP binding data to identify potential 
    transcription factor binding sites. They use gene expression data from 
    sequencing or microarrays to quantify the effect of the factor 
    overexpression or knockdown on its targets. The integration of the binding 
    and expression data therefore can be used to improve the understanding of a 
    transcription factor function. In this workshop, we present a complete 
    workflow for integrating the gene expression (RNA-seq) and DNA-binding data 
    (ChIP-seq) to predict the combined function of two transcription factors 
    using R/Bioconductor. The example we will be using in the workshop is from 
    real datasets of two functionally and evolutionary related transcription 
    factors YY1 and YY2 in HeLa cells. We will try to identify the 
    factor-specific and the shared targets of the factors in this particular 
    cell line. Then we will use a technique find out the aggregate functions of 
    the factors on their individual (inducer or repressor) and common targets
    (cooperative or competitive).">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-93043521-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-93043521-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">IntegratingDataBioc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/workshop_code.html">Code Walkthrough: A use case of YY1 and YY2 in HeLa cells</a>
    </li>
    <li>
      <a href="articles/workshop_introduction.html">Introduction: Introduction to target package</a>
    </li>
    <li>
      <a href="articles/workshop_packages.html">Introduction: Bioconductor packages and classes</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/MahShaaban/IntegratingDataBioc">
    <span class="fas fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">

<p><img src="https://github.com/MahShaaban/IntegratingDataBioc/workflows/.github/workflows/basic_checks.yaml/badge.svg" alt=".github/workflows/basic_checks.yaml"></p>
<ul>
<li>This workshop is scheduled for February, 27 and is hosted by <a href="https://rladies.org/tunisia-rladies/">R-Ladies Tunisia</a> as part of the <a href="R4Bioinfo">R4Bioinfo</a> series.</li>
<li>Part of the materials was presented at <a href="https://github.com/MahShaaban/targetShop">EuropBioc 2021</a>.</li>
<li>The code is organized in a <a href="https://github.com/seandavi/BuildABiocWorkshop">BiocWorkshop</a> format.</li>
</ul>
<div id="integrating-gene-expression-and-dna-binding-data-using-rbioconductor" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#integrating-gene-expression-and-dna-binding-data-using-rbioconductor" class="anchor"></a>Integrating gene expression and DNA-binding data using R/Bioconductor</h1></div>
<div id="description" class="section level2">
<h2 class="hasAnchor">
<a href="#description" class="anchor"></a>Description</h2>
<p>Researchers use ChIP binding data to identify potential transcription factor binding sites. They use gene expression data from sequencing or microarrays to quantify the effect of the factor over-expression or knockdown on its targets. The integration of the binding and expression data therefore can be used to improve the understanding of a transcription factor function. In this workshop, I present a complete workflow for integrating the gene expression (microarray/RNA-seq) and DNA-binding data (ChIP-seq) to predict the combined function of two transcription factors using R/Bioconductor. The example we will be using in the workshop is from real datasets of two functionally and evolutionary related transcription factors YY1 and YY2 in HeLa cells. We will try to identify the factor-specific and the shared targets of the factors in this particular cell line. Then we will use a technique find out the aggregate functions of the factors on their individual (inducer or repressor) and common targets (cooperative or competitive).</p>
</div>
<div id="pre-requisites" class="section level2">
<h2 class="hasAnchor">
<a href="#pre-requisites" class="anchor"></a>Pre-requisites</h2>
<ul>
<li>Basic knowledge of <em>R</em> syntax</li>
<li>Familiarity with gene expression and DNA-binding data</li>
<li>Readings: <a href="https://pubmed.ncbi.nlm.nih.gov/21940749/">Tang et al., 2011</a>, <a href="https://pubmed.ncbi.nlm.nih.gov/24263090/">Wang et al., 2013</a>, and <a href="https://pubmed.ncbi.nlm.nih.gov/32894066/">Ahmed et al., 2020</a>
</li>
</ul>
</div>
<div id="participation" class="section level2">
<h2 class="hasAnchor">
<a href="#participation" class="anchor"></a>Participation</h2>
<p>Participants are expected to walk through the code (rmarkdown document). An introduction will be given at the beginning to introduce the motivation of the analysis and the relevant Bioconductor packages.</p>
<p>There are a few ways to run the code</p>
<ol>
<li><p>Use <a href="https://rstudio.cloud/">RStudio Cloud</a> to run a cloud instance of RStudio (free). Create a New Project &gt; New Project from Git Repository and paste this URL in the pop up box <a href="https://github.com/MahShaaban/IntegratingDataBioc">MahShaaban/IntegratingDataBioc</a></p></li>
<li><p>Use RStudio locally to install the required packages and run the code after cloning this repo</p></li>
</ol>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">git</span> clone https://github.com/MahShaaban/IntegratingDataBioc</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> IntegratingDataBioc</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="ex">open</span> -a Rstudio vignette/workshop_code.Rmd </span></code></pre></div>
<ol>
<li>Use the docker image <a href="https://hub.docker.com/repository/docker/mahshaaban/IntegratingDataBioc/">mahshaaban/IntegratingDataBioc</a> and knit the <code>Rmd</code> files in <code>vignettes/</code> from within Rstudio.</li>
</ol>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="ex">docker</span> pull mahshaaban/target:latest</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="ex">docker</span> run -e PASSWORD=<span class="op">&lt;</span>a_password<span class="op">&gt;</span> -p 8787:8787 mahshaaban/target:latest</span></code></pre></div>
<p>An Rstudio session will be accessable at <a href="https://localhost:8787/">https://localhost:8787/</a> in the browser. The login username is always <code>rstudio</code> and the password is <code>&lt;a_password&gt;</code>.</p>
</div>
<div id="r--bioconductor-packages-used" class="section level2">
<h2 class="hasAnchor">
<a href="#r--bioconductor-packages-used" class="anchor"></a><em>R</em> / <em>Bioconductor</em> packages used</h2>
<ul>
<li>target</li>
</ul>
<p>Data management</p>
<ul>
<li>GenomicRanges</li>
<li>GenomicFeatures</li>
<li>rtracklayer</li>
<li>AnnotationDbi</li>
<li>readr</li>
<li>dplyr</li>
<li>tidyr</li>
<li>purrr</li>
</ul>
<p>Annotation packages</p>
<ul>
<li>TxDb.Hsapiens.UCSC.hg19.knownGene</li>
<li>org.Hs.eg.db</li>
</ul>
</div>
<div id="time-outline" class="section level2">
<h2 class="hasAnchor">
<a href="#time-outline" class="anchor"></a>Time outline</h2>
<table class="table">
<thead><tr class="header">
<th>Activity</th>
<th>Time</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Bioconductor packages and classes</td>
<td>20m</td>
</tr>
<tr class="even">
<td>Introduction to the target package</td>
<td>20m</td>
</tr>
<tr class="odd">
<td>Code-walkthrough: A use case of YY1 and YY2 in HeLa cells</td>
<td>20m</td>
</tr>
</tbody>
</table>
</div>
<div id="workshop-goals-and-objectives" class="section level2">
<h2 class="hasAnchor">
<a href="#workshop-goals-and-objectives" class="anchor"></a>Workshop goals and objectives</h2>
<p>The workshop aims to teach participants how to use R/Bioconductor packages to read in differential expression and binding peaks data, run a predictive analysis and explore its output. I hope that by providing a complete realistic example, participants would develop an understanding of the issues and the importance of integrating those two types of data. Ideally, participants would be able to adapt this code and the workflow to apply this kind of analysis to their own datasets.</p>
</div>
<div id="learning-goals" class="section level2">
<h2 class="hasAnchor">
<a href="#learning-goals" class="anchor"></a>Learning goals</h2>
<ul>
<li>Learn to read differential expression and binding peaks data into the appropriate R objects</li>
<li>Learn to use Bioconductor packages to extract the genomic annotation</li>
<li>Learn to prepare the expression and binding data for <code>target</code> analysis</li>
<li>Understand the <code>target</code> output through the package visualization and testing tools</li>
</ul>
</div>
<div id="learning-objectives" class="section level2">
<h2 class="hasAnchor">
<a href="#learning-objectives" class="anchor"></a>Learning objectives</h2>
<ul>
<li>Read data into R <code>data.frame</code>s and Bioconductor <code>GRanges</code> objects</li>
<li>Extract information from Bioconductor annotation packages TxDb and org.db</li>
<li>Apply the <code>target</code> analysis using <code>associated_peaks</code> and <code>direct_targets</code> functions</li>
<li>Visualize the output using the cumulative distribution functions through <code>plot_predicitons</code>
</li>
<li>Test the results using KS test through <code>test_predicitons</code>
</li>
</ul>
<p>This workshop is based on a workflow article: <a href="https://github.com/MahShaaban/targetFlow">a draft</a></p>
</div>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Browse source code at <br><a href="https://github.com/MahShaaban/IntegratingDataBioc/">https://​github.com/​MahShaaban/​IntegratingDataBioc/​</a>
</li>
<li>Report a bug at <br><a href="https://github.com/MahShaaban/IntegratingDataBioc/issues/new/choose">https://​github.com/​MahShaaban/​IntegratingDataBioc/​issues/​new/​choose</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Mahmoud Ahmed <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-4377-6541" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p>Developed by Mahmoud Ahmed.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
