<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Code Walkthrough: A use case of YY1 and YY2 in HeLa cells • IntegratingDataBioc</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Code Walkthrough: A use case of YY1 and YY2 in HeLa cells">
<meta property="og:description" content="IntegratingDataBioc">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-93043521-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-93043521-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">IntegratingDataBioc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/workshop_code.html">Code Walkthrough: A use case of YY1 and YY2 in HeLa cells</a>
    </li>
    <li>
      <a href="../articles/workshop_introduction.html">Introduction: Introduction to target package</a>
    </li>
    <li>
      <a href="../articles/workshop_packages.html">Introduction: Bioconductor packages and classes</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/MahShaaban/IntegratingDataBioc">
    <span class="fas fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="workshop_code_files/header-attrs-2.6/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Code Walkthrough: A use case of YY1 and YY2 in HeLa cells</h1>
                        <h4 class="author">Mahmoud Ahmed<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>
</h4>
            
            <h4 class="date">2021-02-18</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/MahShaaban/IntegratingDataBioc/blob/master/vignettes/workshop_code.Rmd"><code>vignettes/workshop_code.Rmd</code></a></small>
      <div class="hidden name"><code>workshop_code.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># load required libraries</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org">tidyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://readr.tidyverse.org">readr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://purrr.tidyverse.org">purrr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicRanges">GenomicRanges</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicFeatures">GenomicFeatures</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">rtracklayer</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/AnnotationDbi">AnnotationDbi</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">TxDb.Hsapiens.UCSC.hg19.knownGene</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/MahShaaban/target">target</a></span><span class="op">)</span></code></pre></div>
<div id="motivation" class="section level1">
<h1 class="hasAnchor">
<a href="#motivation" class="anchor"></a>Motivation</h1>
<p>YY1 and YY2 belongs to the same family of transcription factors.</p>
<ul>
<li>
<p>Ying Yang 1 (YY1)</p>
<ul>
<li>A zinc finger protein</li>
<li>Direct deacetylase and histone acetyltransferases of many promoters</li>
<li>Induces or represses the expression of the target genes</li>
</ul>
</li>
<li>
<p>Ying Yang 2 (YY2)</p>
<ul>
<li>A zinc finger protein</li>
<li>Arose by retro-transposition of YY1</li>
</ul>
</li>
</ul>
<p>Using the target analysis, we will attempt to answer the following questions:</p>
<ul>
<li>Do the two transcription factors share the same target genes?</li>
<li>What are the consequences of the binding of each factor on its targets?</li>
<li>On the shared targets, how the two factors functionally interact?</li>
</ul>
</div>
<div id="datasets" class="section level1">
<h1 class="hasAnchor">
<a href="#datasets" class="anchor"></a>Datasets</h1>
<p>To answer these questions, we use publicly available datasets.</p>
<p>Table 1. Expression and binding data of YY1 and YY2 in HeLa cells.</p>
<table class="table">
<thead><tr class="header">
<th>GEO ID</th>
<th>Data Type</th>
<th>Design</th>
<th>Ref.</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>GSE14964</td>
<td>Microarrays</td>
<td>YY#-knockdown</td>
<td><span class="citation">Chen et al. (2010)</span></td>
</tr>
<tr class="even">
<td>GSE31417</td>
<td>ChIP-Seq</td>
<td>YY1 vs input</td>
<td><span class="citation">Michaud et al. (2013)</span></td>
</tr>
<tr class="odd">
<td>GSE96878</td>
<td>ChIP-Seq</td>
<td>YY2 vs input</td>
<td><span class="citation">Wu et al. (2017)</span></td>
</tr>
</tbody>
</table>
<p>Data pre-processing:</p>
<ul>
<li>Microarrays were obtained in the form of differential expression between the two conditions from <a href="http://www.licpathway.net/KnockTF/index.html">KnockTF</a>.</li>
<li>The ChIP peaks were obtained in the form of <code>bed</code> files from <a href="https://chip-atlas.org">ChIP-Atlas</a>.</li>
<li>USSC hg19 human genome to extract the genomic annotation</li>
</ul>
<p>Data analysis:</p>
<ol style="list-style-type: decimal">
<li>Prepare the three sources of data for the target analysis</li>
<li>Predict the specific targets for each individual factors</li>
<li>Predict the combined function of the two factors on the shared target genes</li>
</ol>
<div id="preparing-the-binding-data" class="section level2">
<h2 class="hasAnchor">
<a href="#preparing-the-binding-data" class="anchor"></a>Preparing the binding data</h2>
<p>The ChIP peaks were downloaded in the form of separate bed files for each factor. We first locate the files in the <code>data/</code> directory and load the files using <code>import.bed</code>. Then the data is transformed into a suitable format, <code>GRanges</code>. The resulting object, <code>peaks</code>, is a <code>list</code> of two <code>GRanges</code> items, one for each factor.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># locate the peaks bed files</span>
<span class="va">peak_files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>YY1 <span class="op">=</span> <span class="st">'data/Oth.Utr.05.YY1.AllCell.bed'</span>,
                YY2 <span class="op">=</span> <span class="st">'data/Oth.Utr.05.YY2.AllCell.bed'</span><span class="op">)</span>

<span class="co"># load the peaks bed files as GRanges</span>
<span class="va">peaks</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map</a></span><span class="op">(</span><span class="va">peak_files</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/rtracklayer/man/BEDFile-class.html">import.bed</a></span><span class="op">(</span><span class="va">.x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># show the numbers of peaks of the two factors</span>
<span class="fu"><a href="https://rdrr.io/r/base/lengths.html">lengths</a></span><span class="op">(</span><span class="va">peaks</span><span class="op">)</span>
<span class="co">#&gt;   YY1   YY2 </span>
<span class="co">#&gt; 31490  3368</span>

<span class="co"># show the first few entries of the GRanges object</span>
<span class="fu">show</span><span class="op">(</span><span class="va">peaks</span><span class="op">$</span><span class="va">YY1</span><span class="op">)</span>
<span class="co">#&gt; GRanges object with 31490 ranges and 4 metadata columns:</span>
<span class="co">#&gt;           seqnames            ranges strand |                   name     score</span>
<span class="co">#&gt;              &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |            &lt;character&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;       [1]     chr1        9980-10149      * | ID=SRX092576;Name=YY..       186</span>
<span class="co">#&gt;       [2]     chr1       10001-10265      * | ID=SRX190209;Name=YY..       512</span>
<span class="co">#&gt;       [3]     chr1       10368-10452      * | ID=SRX190209;Name=YY..       128</span>
<span class="co">#&gt;       [4]     chr1       11219-11322      * | ID=SRX190209;Name=YY..       100</span>
<span class="co">#&gt;       [5]     chr1       29140-29432      * | ID=SRX190209;Name=YY..       418</span>
<span class="co">#&gt;       ...      ...               ...    ... .                    ...       ...</span>
<span class="co">#&gt;   [31486]     chrY 58996509-58996690      * | ID=SRX092576;Name=YY..       135</span>
<span class="co">#&gt;   [31487]     chrY 58996927-58997108      * | ID=SRX092576;Name=YY..       183</span>
<span class="co">#&gt;   [31488]     chrY 58996934-58997092      * | ID=SRX190209;Name=YY..       111</span>
<span class="co">#&gt;   [31489]     chrY 59363025-59363426      * | ID=SRX190209;Name=YY..       894</span>
<span class="co">#&gt;   [31490]     chrY 59363027-59363397      * | ID=SRX092576;Name=YY..       404</span>
<span class="co">#&gt;               itemRgb             thick</span>
<span class="co">#&gt;           &lt;character&gt;         &lt;IRanges&gt;</span>
<span class="co">#&gt;       [1]     #00BDFF        9980-10149</span>
<span class="co">#&gt;       [2]     #0CFF00       10001-10265</span>
<span class="co">#&gt;       [3]     #0082FF       10368-10452</span>
<span class="co">#&gt;       [4]     #0066FF       11219-11322</span>
<span class="co">#&gt;       [5]     #00FF53       29140-29432</span>
<span class="co">#&gt;       ...         ...               ...</span>
<span class="co">#&gt;   [31486]     #0089FF 58996509-58996690</span>
<span class="co">#&gt;   [31487]     #00BAFF 58996927-58997108</span>
<span class="co">#&gt;   [31488]     #0071FF 58996934-58997092</span>
<span class="co">#&gt;   [31489]     #FF6C00 59363025-59363426</span>
<span class="co">#&gt;   [31490]     #00FF61 59363027-59363397</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 56 sequences from an unspecified genome; no seqlengths</span>

<span class="co"># show number of peaks in each chromosome</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="fu">seqnames</span><span class="op">(</span><span class="va">peaks</span><span class="op">$</span><span class="va">YY1</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;                  chr1                 chr10                 chr11 </span>
<span class="co">#&gt;                  3126                  1632                  1590 </span>
<span class="co">#&gt;                 chr12                 chr13                 chr14 </span>
<span class="co">#&gt;                  1817                   605                   971 </span>
<span class="co">#&gt;                 chr15                 chr16                 chr17 </span>
<span class="co">#&gt;                   964                  1246                  1613 </span>
<span class="co">#&gt; chr17_gl000204_random chr17_gl000205_random                 chr18 </span>
<span class="co">#&gt;                     1                    12                   548 </span>
<span class="co">#&gt;                 chr19 chr19_gl000208_random  chr1_gl000191_random </span>
<span class="co">#&gt;                  1626                    18                     4 </span>
<span class="co">#&gt;  chr1_gl000192_random                  chr2                 chr20 </span>
<span class="co">#&gt;                     4                  2304                   775 </span>
<span class="co">#&gt;                 chr21                 chr22                  chr3 </span>
<span class="co">#&gt;                   332                   728                  1935 </span>
<span class="co">#&gt;                  chr4  chr4_gl000193_random  chr4_gl000194_random </span>
<span class="co">#&gt;                  1369                     1                     1 </span>
<span class="co">#&gt;                  chr5                  chr6                  chr7 </span>
<span class="co">#&gt;                  1479                  1688                  1538 </span>
<span class="co">#&gt;  chr7_gl000195_random                  chr8                  chr9 </span>
<span class="co">#&gt;                    11                  1134                  1330 </span>
<span class="co">#&gt;  chr9_gl000198_random  chr9_gl000199_random                  chrM </span>
<span class="co">#&gt;                     1                    37                    10 </span>
<span class="co">#&gt;        chrUn_gl000211        chrUn_gl000212        chrUn_gl000214 </span>
<span class="co">#&gt;                     3                     4                     9 </span>
<span class="co">#&gt;        chrUn_gl000216        chrUn_gl000218        chrUn_gl000219 </span>
<span class="co">#&gt;                    44                     1                     5 </span>
<span class="co">#&gt;        chrUn_gl000220        chrUn_gl000221        chrUn_gl000222 </span>
<span class="co">#&gt;                    37                     1                     5 </span>
<span class="co">#&gt;        chrUn_gl000223        chrUn_gl000224        chrUn_gl000225 </span>
<span class="co">#&gt;                     4                     8                    51 </span>
<span class="co">#&gt;        chrUn_gl000226        chrUn_gl000227        chrUn_gl000228 </span>
<span class="co">#&gt;                    18                     1                    12 </span>
<span class="co">#&gt;        chrUn_gl000235        chrUn_gl000237        chrUn_gl000240 </span>
<span class="co">#&gt;                     3                     1                     1 </span>
<span class="co">#&gt;        chrUn_gl000242        chrUn_gl000243        chrUn_gl000245 </span>
<span class="co">#&gt;                     1                     1                     1 </span>
<span class="co">#&gt;                  chrX                  chrY </span>
<span class="co">#&gt;                   630                   199</span>

<span class="co"># show the the width of the peaks</span>
<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="fu">width</span><span class="op">(</span><span class="va">peaks</span><span class="op">$</span><span class="va">YY1</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt;    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span>
<span class="co">#&gt;    85.0   111.0   151.0   192.9   230.0  1772.0</span>

<span class="co"># show the strands of the peaks</span>
<span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu">strand</span><span class="op">(</span><span class="va">peaks</span><span class="op">$</span><span class="va">YY1</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] *</span>
<span class="co">#&gt; Levels: + - *</span></code></pre></div>
</div>
<div id="preparing-the-expression-data" class="section level2">
<h2 class="hasAnchor">
<a href="#preparing-the-expression-data" class="anchor"></a>Preparing the expression data</h2>
<p>The differential expression data were downloaded in tabular format.</p>
<ul>
<li>Locate the files in <code>data/</code>
</li>
<li>Read the files using <code>read_tsv</code>
</li>
<li>Select and rename the relevant columns</li>
</ul>
<p>The resulting object, <code>express</code>, is a <code>list</code> of two <code>tibble</code> items.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># locate the expression text files</span>
<span class="va">expression_files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>YY1 <span class="op">=</span> <span class="st">'data/DataSet_01_18.tsv'</span>,
                      YY2 <span class="op">=</span> <span class="st">'data/DataSet_01_19.tsv'</span><span class="op">)</span>

<span class="co"># load the expression text files</span>
<span class="va">express</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map</a></span><span class="op">(</span><span class="va">expression_files</span>,
               <span class="op">~</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html">read_tsv</a></span><span class="op">(</span><span class="va">.x</span>, col_names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span>
                 <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">7</span>, <span class="fl">9</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="co">#9</span>
                 <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html">setNames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'tf'</span>, <span class="st">'gene'</span>, <span class="st">'fc'</span>, <span class="st">'pvalue'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
                 <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">tf</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'YY1'</span>, <span class="st">'YY2'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
                 <span class="fu"><a href="https://rdrr.io/r/stats/na.fail.html">na.omit</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># show the number of genes with recorded fold-change and p-value</span>
<span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map</a></span><span class="op">(</span><span class="va">express</span>, <span class="va">nrow</span><span class="op">)</span>
<span class="co">#&gt; $YY1</span>
<span class="co">#&gt; [1] 16595</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $YY2</span>
<span class="co">#&gt; [1] 16595</span>

<span class="co"># show same genes were recorded for both factors</span>
<span class="fu"><a href="https://rdrr.io/r/base/all.html">all</a></span><span class="op">(</span><span class="va">express</span><span class="op">$</span><span class="va">YY1</span><span class="op">$</span><span class="va">gene</span> <span class="op">%in%</span> <span class="va">express</span><span class="op">$</span><span class="va">YY2</span><span class="op">$</span><span class="va">gene</span><span class="op">)</span>
<span class="co">#&gt; [1] TRUE</span>

<span class="co"># show direction and significance of the regulation</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">express</span><span class="op">$</span><span class="va">YY1</span><span class="op">$</span><span class="va">fc</span> <span class="op">&gt;</span> <span class="fl">0</span>, <span class="va">express</span><span class="op">$</span><span class="va">YY1</span><span class="op">$</span><span class="va">pvalue</span> <span class="op">&lt;</span> <span class="fl">.01</span><span class="op">)</span>
<span class="co">#&gt;        </span>
<span class="co">#&gt;         FALSE TRUE</span>
<span class="co">#&gt;   FALSE  6785  160</span>
<span class="co">#&gt;   TRUE   9280  370</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">express</span><span class="op">$</span><span class="va">YY2</span><span class="op">$</span><span class="va">fc</span> <span class="op">&gt;</span> <span class="fl">0</span>, <span class="va">express</span><span class="op">$</span><span class="va">YY2</span><span class="op">$</span><span class="va">pvalue</span> <span class="op">&lt;</span> <span class="fl">.01</span><span class="op">)</span>
<span class="co">#&gt;        </span>
<span class="co">#&gt;         FALSE TRUE</span>
<span class="co">#&gt;   FALSE  7435  132</span>
<span class="co">#&gt;   TRUE   8804  224</span></code></pre></div>
<p>The knockdown of either factor in HeLa cells seem to change the expression of many genes in either directions (Figure 1A&amp;B). Moreover, the changes resulting from the knockdown of the factors individually are correlated (Figure 1C). This suggest that, many of the regulated genes are shared targets of the two factors or they respond similarly to the perturbation of either factor.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Figure 1</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span>
<span class="co"># volcano plot of YY1 knockdown</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">express</span><span class="op">$</span><span class="va">YY1</span><span class="op">$</span><span class="va">fc</span>, 
     <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="va">express</span><span class="op">$</span><span class="va">YY1</span><span class="op">$</span><span class="va">pvalue</span><span class="op">)</span>,
     xlab <span class="op">=</span> <span class="st">'Fold-change (log_2)'</span>,
     ylab <span class="op">=</span> <span class="st">'P-value (-log_10)'</span>,
     xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">4</span>, <span class="fl">4</span><span class="op">)</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">6</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/title.html">title</a></span><span class="op">(</span><span class="st">'(A)'</span><span class="op">)</span>

<span class="co"># volcano plot of YY2 knockdown</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">express</span><span class="op">$</span><span class="va">YY2</span><span class="op">$</span><span class="va">fc</span>, 
     <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="va">express</span><span class="op">$</span><span class="va">YY2</span><span class="op">$</span><span class="va">pvalue</span><span class="op">)</span>,
     xlab <span class="op">=</span> <span class="st">'Fold-change (log_2)'</span>,
     ylab <span class="op">=</span> <span class="st">'P-value (-log_10)'</span>,
     xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">4</span>, <span class="fl">4</span><span class="op">)</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">6</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/title.html">title</a></span><span class="op">(</span><span class="st">'(B)'</span><span class="op">)</span>

<span class="co"># plot fold-change of YY1 and YY2</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">express</span><span class="op">$</span><span class="va">YY1</span><span class="op">$</span><span class="va">fc</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">express</span><span class="op">$</span><span class="va">YY1</span><span class="op">$</span><span class="va">gene</span><span class="op">)</span><span class="op">]</span>,
     <span class="va">express</span><span class="op">$</span><span class="va">YY2</span><span class="op">$</span><span class="va">fc</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">express</span><span class="op">$</span><span class="va">YY2</span><span class="op">$</span><span class="va">gene</span><span class="op">)</span><span class="op">]</span>,
     xlab <span class="op">=</span> <span class="st">'YY1-knockdown (log_2)'</span>,
     ylab <span class="op">=</span> <span class="st">'YY2-knockdown (log_2)'</span>,
     xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">4</span>, <span class="fl">4</span><span class="op">)</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">4</span>, <span class="fl">4</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/title.html">title</a></span><span class="op">(</span><span class="st">'(C)'</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="workshop_code_files/figure-html/foldchange-1.png" alt="Figure 1. Differential expression between factor knockdown and control HeLa cells. Gene expression was compared between transcription factors knockdown and control HeLa cells. The fold-change and p-values of (A) YY1- and (B) YY2-knockdown are shown as volcano plots. (C) Scatter plot of the fold-change of the YY1- and YY2-knockdown." width="100%"><p class="caption">
Figure 1. Differential expression between factor knockdown and control HeLa cells. Gene expression was compared between transcription factors knockdown and control HeLa cells. The fold-change and p-values of (A) YY1- and (B) YY2-knockdown are shown as volcano plots. (C) Scatter plot of the fold-change of the YY1- and YY2-knockdown.
</p>
</div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># show the correlation between the fold-changes</span>
<span class="fu"><a href="https://rdrr.io/r/stats/cor.html">cor</a></span><span class="op">(</span>
  <span class="va">express</span><span class="op">$</span><span class="va">YY1</span><span class="op">$</span><span class="va">fc</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">express</span><span class="op">$</span><span class="va">YY1</span><span class="op">$</span><span class="va">gene</span><span class="op">)</span><span class="op">]</span>,
  <span class="va">express</span><span class="op">$</span><span class="va">YY2</span><span class="op">$</span><span class="va">fc</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">express</span><span class="op">$</span><span class="va">YY2</span><span class="op">$</span><span class="va">gene</span><span class="op">)</span><span class="op">]</span>,
<span class="op">)</span>
<span class="co">#&gt; [1] 0.5632793</span></code></pre></div>
</div>
<div id="preparing-genome-annotation" class="section level2">
<h2 class="hasAnchor">
<a href="#preparing-genome-annotation" class="anchor"></a>Preparing genome annotation</h2>
<p>The gene information in <code>express</code> are recorded using the gene SYMBOLS.</p>
<ul>
<li>Map SYMBOLS to the ENTREZIDs using <code>org.Hs.eg.db</code>
</li>
<li>Extract the genomic coordinates from <code>TxDb.Hsapiens.UCSC.hg19.knownGene</code>
</li>
<li>Resize the transcripts to 100kb upstream from transcription start sites</li>
</ul>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># map symbols to entrez ids</span>
<span class="va">symbol_entrez</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html">select</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span>,
                                       keys <span class="op">=</span> <span class="va">express</span><span class="op">$</span><span class="va">YY1</span><span class="op">$</span><span class="va">gene</span>,
                                       columns <span class="op">=</span> <span class="st">'ENTREZID'</span>,
                                       keytype <span class="op">=</span> <span class="st">'SYMBOL'</span><span class="op">)</span>

<span class="co"># remove unmapped genes</span>
<span class="va">symbol_entrez</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/na.fail.html">na.omit</a></span><span class="op">(</span><span class="va">symbol_entrez</span><span class="op">)</span>

<span class="co"># resize regions</span>
<span class="va">genome</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html">promoters</a></span><span class="op">(</span><span class="va">TxDb.Hsapiens.UCSC.hg19.knownGene</span>,
                    upstream <span class="op">=</span> <span class="fl">100000</span>,
                    filter <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>gene_id <span class="op">=</span> <span class="va">symbol_entrez</span><span class="op">$</span><span class="va">ENTREZID</span><span class="op">)</span>,
                    columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'tx_id'</span>, <span class="st">'tx_name'</span>, <span class="st">'gene_id'</span><span class="op">)</span><span class="op">)</span>

<span class="co"># match and add symbols</span>
<span class="va">ind</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span><span class="op">(</span><span class="va">genome</span><span class="op">$</span><span class="va">gene_id</span><span class="op">@</span><span class="va">unlistData</span>, <span class="va">symbol_entrez</span><span class="op">$</span><span class="va">ENTREZID</span><span class="op">)</span>
<span class="va">genome</span><span class="op">$</span><span class="va">gene</span> <span class="op">&lt;-</span> <span class="va">symbol_entrez</span><span class="op">$</span><span class="va">SYMBOL</span><span class="op">[</span><span class="va">ind</span><span class="op">]</span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># show the first few entries of the mapping data.frame</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">symbol_entrez</span><span class="op">)</span>
<span class="co">#&gt;    SYMBOL ENTREZID</span>
<span class="co">#&gt; 1  IFI44L    10964</span>
<span class="co">#&gt; 2 C7orf57   136288</span>
<span class="co">#&gt; 3  KLHL32   114792</span>
<span class="co">#&gt; 4   TREM2    54209</span>
<span class="co">#&gt; 5   EPHX2     2053</span>
<span class="co">#&gt; 6  LRRC25   126364</span>

<span class="co"># show txdb object</span>
<span class="va">TxDb.Hsapiens.UCSC.hg19.knownGene</span>
<span class="co">#&gt; TxDb object:</span>
<span class="co">#&gt; # Db type: TxDb</span>
<span class="co">#&gt; # Supporting package: GenomicFeatures</span>
<span class="co">#&gt; # Data source: UCSC</span>
<span class="co">#&gt; # Genome: hg19</span>
<span class="co">#&gt; # Organism: Homo sapiens</span>
<span class="co">#&gt; # Taxonomy ID: 9606</span>
<span class="co">#&gt; # UCSC Table: knownGene</span>
<span class="co">#&gt; # Resource URL: http://genome.ucsc.edu/</span>
<span class="co">#&gt; # Type of Gene ID: Entrez Gene ID</span>
<span class="co">#&gt; # Full dataset: yes</span>
<span class="co">#&gt; # miRBase build ID: GRCh37</span>
<span class="co">#&gt; # transcript_nrow: 82960</span>
<span class="co">#&gt; # exon_nrow: 289969</span>
<span class="co">#&gt; # cds_nrow: 237533</span>
<span class="co">#&gt; # Db created by: GenomicFeatures package from Bioconductor</span>
<span class="co">#&gt; # Creation time: 2015-10-07 18:11:28 +0000 (Wed, 07 Oct 2015)</span>
<span class="co">#&gt; # GenomicFeatures version at creation time: 1.21.30</span>
<span class="co">#&gt; # RSQLite version at creation time: 1.0.0</span>
<span class="co">#&gt; # DBSCHEMAVERSION: 1.1</span>

<span class="co"># show the</span>
<span class="fu">show</span><span class="op">(</span><span class="va">genome</span><span class="op">)</span>
<span class="co">#&gt; GRanges object with 56521 ranges and 4 metadata columns:</span>
<span class="co">#&gt;                    seqnames        ranges strand |     tx_id     tx_name</span>
<span class="co">#&gt;                       &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;</span>
<span class="co">#&gt;   uc001abv.1           chr1 760530-860729      + |        21  uc001abv.1</span>
<span class="co">#&gt;   uc001abw.1           chr1 761121-861320      + |        22  uc001abw.1</span>
<span class="co">#&gt;   uc031pjl.1           chr1 761302-861501      + |        23  uc031pjl.1</span>
<span class="co">#&gt;   uc031pjm.1           chr1 761302-861501      + |        24  uc031pjm.1</span>
<span class="co">#&gt;   uc031pjn.1           chr1 761302-861501      + |        25  uc031pjn.1</span>
<span class="co">#&gt;          ...            ...           ...    ... .       ...         ...</span>
<span class="co">#&gt;   uc011mgb.1 chrUn_gl000223  68588-168787      - |     82934  uc011mgb.1</span>
<span class="co">#&gt;   uc011mgc.1 chrUn_gl000223  68588-168787      - |     82935  uc011mgc.1</span>
<span class="co">#&gt;   uc011mgd.2 chrUn_gl000223 119531-219730      - |     82936  uc011mgd.2</span>
<span class="co">#&gt;   uc031tgg.1 chrUn_gl000223 119531-219730      - |     82937  uc031tgg.1</span>
<span class="co">#&gt;   uc031tgh.1 chrUn_gl000223 119531-219730      - |     82938  uc031tgh.1</span>
<span class="co">#&gt;                      gene_id        gene</span>
<span class="co">#&gt;              &lt;CharacterList&gt; &lt;character&gt;</span>
<span class="co">#&gt;   uc001abv.1          148398      SAMD11</span>
<span class="co">#&gt;   uc001abw.1          148398      SAMD11</span>
<span class="co">#&gt;   uc031pjl.1          148398      SAMD11</span>
<span class="co">#&gt;   uc031pjm.1          148398      SAMD11</span>
<span class="co">#&gt;   uc031pjn.1          148398      SAMD11</span>
<span class="co">#&gt;          ...             ...         ...</span>
<span class="co">#&gt;   uc011mgb.1            7637       ZNF84</span>
<span class="co">#&gt;   uc011mgc.1            7637       ZNF84</span>
<span class="co">#&gt;   uc011mgd.2            7574       ZNF26</span>
<span class="co">#&gt;   uc031tgg.1            7574       ZNF26</span>
<span class="co">#&gt;   uc031tgh.1            7574       ZNF26</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 93 sequences (1 circular) from hg19 genome</span></code></pre></div>
<p>Now the two objects can be merged. The merged object, <code>regions</code>, is similarly a <code>GRanges</code> that contains genome and expression information of all common genes.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># make regions by merging the genome and express data</span>
<span class="va">regions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map</a></span><span class="op">(</span><span class="va">express</span>,
               <span class="op">~</span><span class="op">{</span>
                 <span class="co"># make a copy of genome</span>
                 <span class="va">gr</span> <span class="op">&lt;-</span> <span class="va">genome</span>
                 
                 <span class="co"># match gene names </span>
                 <span class="va">ind</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span><span class="op">(</span><span class="va">gr</span><span class="op">$</span><span class="va">gene</span>, <span class="va">.x</span><span class="op">$</span><span class="va">gene</span><span class="op">)</span>
                 
                 <span class="co"># add the expression info to the metadata</span>
                 <span class="fu">mcols</span><span class="op">(</span><span class="va">gr</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">gr</span><span class="op">)</span>, <span class="va">.x</span><span class="op">[</span><span class="va">ind</span>,<span class="op">]</span><span class="op">)</span>
                 <span class="va">gr</span>
               <span class="op">}</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># show the first few entries in the GRanges object</span>
<span class="va">regions</span><span class="op">$</span><span class="va">YY1</span>
<span class="co">#&gt; GRanges object with 56521 ranges and 8 metadata columns:</span>
<span class="co">#&gt;                    seqnames        ranges strand |     tx_id     tx_name</span>
<span class="co">#&gt;                       &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;</span>
<span class="co">#&gt;   uc001abv.1           chr1 760530-860729      + |        21  uc001abv.1</span>
<span class="co">#&gt;   uc001abw.1           chr1 761121-861320      + |        22  uc001abw.1</span>
<span class="co">#&gt;   uc031pjl.1           chr1 761302-861501      + |        23  uc031pjl.1</span>
<span class="co">#&gt;   uc031pjm.1           chr1 761302-861501      + |        24  uc031pjm.1</span>
<span class="co">#&gt;   uc031pjn.1           chr1 761302-861501      + |        25  uc031pjn.1</span>
<span class="co">#&gt;          ...            ...           ...    ... .       ...         ...</span>
<span class="co">#&gt;   uc011mgb.1 chrUn_gl000223  68588-168787      - |     82934  uc011mgb.1</span>
<span class="co">#&gt;   uc011mgc.1 chrUn_gl000223  68588-168787      - |     82935  uc011mgc.1</span>
<span class="co">#&gt;   uc011mgd.2 chrUn_gl000223 119531-219730      - |     82936  uc011mgd.2</span>
<span class="co">#&gt;   uc031tgg.1 chrUn_gl000223 119531-219730      - |     82937  uc031tgg.1</span>
<span class="co">#&gt;   uc031tgh.1 chrUn_gl000223 119531-219730      - |     82938  uc031tgh.1</span>
<span class="co">#&gt;                      gene_id        gene          tf        gene        fc</span>
<span class="co">#&gt;              &lt;CharacterList&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   uc001abv.1          148398      SAMD11         YY1      SAMD11  -0.54365</span>
<span class="co">#&gt;   uc001abw.1          148398      SAMD11         YY1      SAMD11  -0.54365</span>
<span class="co">#&gt;   uc031pjl.1          148398      SAMD11         YY1      SAMD11  -0.54365</span>
<span class="co">#&gt;   uc031pjm.1          148398      SAMD11         YY1      SAMD11  -0.54365</span>
<span class="co">#&gt;   uc031pjn.1          148398      SAMD11         YY1      SAMD11  -0.54365</span>
<span class="co">#&gt;          ...             ...         ...         ...         ...       ...</span>
<span class="co">#&gt;   uc011mgb.1            7637       ZNF84         YY1       ZNF84   0.08999</span>
<span class="co">#&gt;   uc011mgc.1            7637       ZNF84         YY1       ZNF84   0.08999</span>
<span class="co">#&gt;   uc011mgd.2            7574       ZNF26         YY1       ZNF26  -0.47400</span>
<span class="co">#&gt;   uc031tgg.1            7574       ZNF26         YY1       ZNF26  -0.47400</span>
<span class="co">#&gt;   uc031tgh.1            7574       ZNF26         YY1       ZNF26  -0.47400</span>
<span class="co">#&gt;                 pvalue</span>
<span class="co">#&gt;              &lt;numeric&gt;</span>
<span class="co">#&gt;   uc001abv.1   0.05581</span>
<span class="co">#&gt;   uc001abw.1   0.05581</span>
<span class="co">#&gt;   uc031pjl.1   0.05581</span>
<span class="co">#&gt;   uc031pjm.1   0.05581</span>
<span class="co">#&gt;   uc031pjn.1   0.05581</span>
<span class="co">#&gt;          ...       ...</span>
<span class="co">#&gt;   uc011mgb.1   0.68511</span>
<span class="co">#&gt;   uc011mgc.1   0.68511</span>
<span class="co">#&gt;   uc011mgd.2   0.09301</span>
<span class="co">#&gt;   uc031tgg.1   0.09301</span>
<span class="co">#&gt;   uc031tgh.1   0.09301</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 93 sequences (1 circular) from hg19 genome</span>

<span class="co"># show the names of the columns in the object metadata</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">regions</span><span class="op">$</span><span class="va">YY1</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] "tx_id"   "tx_name" "gene_id" "gene"    "tf"      "gene"    "fc"     </span>
<span class="co">#&gt; [8] "pvalue"</span>

<span class="co"># show the width of the regions</span>
<span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu">width</span><span class="op">(</span><span class="va">regions</span><span class="op">$</span><span class="va">YY1</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] 100200</span></code></pre></div>
</div>
</div>
<div id="predicting-gene-targets-of-individual-factors" class="section level1">
<h1 class="hasAnchor">
<a href="#predicting-gene-targets-of-individual-factors" class="anchor"></a>Predicting gene targets of individual factors</h1>
<p>The standard target analysis includes the identification of associated peaks using <code>associated_peaks</code> and direct targets using <code>direct_targets</code>.</p>
<p>The inputs for these functions are</p>
<ul>
<li><code>peaks</code></li>
<li><code>regions</code></li>
<li>
<code>regions_col</code>, the column names for regions</li>
<li>
<code>stats_col</code>, the statistics column which is the fold-change in this case.</li>
</ul>
<p>The resulting objects are <code>GRanges</code> for the identified peaks assigned to the regions or the ranked targets. Several columns is added to the metadata objects of the <code>GRanges</code> to save the calculations.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># get associated peaks</span>
<span class="va">ap</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map2.html">map2</a></span><span class="op">(</span><span class="va">peaks</span>, <span class="va">regions</span>,
           <span class="op">~</span><span class="fu"><a href="https://rdrr.io/pkg/target/man/associated_peaks.html">associated_peaks</a></span><span class="op">(</span>peaks<span class="op">=</span><span class="va">.x</span>,
                             regions <span class="op">=</span> <span class="va">.y</span>,
                             regions_col <span class="op">=</span> <span class="st">'tx_id'</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># show associated_peak return</span>
<span class="va">ap</span>
<span class="co">#&gt; $YY1</span>
<span class="co">#&gt; GRanges object with 160722 ranges and 7 metadata columns:</span>
<span class="co">#&gt;            seqnames              ranges strand |                   name</span>
<span class="co">#&gt;               &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |            &lt;character&gt;</span>
<span class="co">#&gt;        [1]     chr1       762751-762895      * | ID=SRX190209;Name=YY..</span>
<span class="co">#&gt;        [2]     chr1       762751-762895      * | ID=SRX190209;Name=YY..</span>
<span class="co">#&gt;        [3]     chr1       762751-762895      * | ID=SRX190209;Name=YY..</span>
<span class="co">#&gt;        [4]     chr1       762751-762895      * | ID=SRX190209;Name=YY..</span>
<span class="co">#&gt;        [5]     chr1       762751-762895      * | ID=SRX190209;Name=YY..</span>
<span class="co">#&gt;        ...      ...                 ...    ... .                    ...</span>
<span class="co">#&gt;   [160718]     chrX 154611695-154611857      * | ID=SRX190209;Name=YY..</span>
<span class="co">#&gt;   [160719]     chrX 154611695-154611857      * | ID=SRX190209;Name=YY..</span>
<span class="co">#&gt;   [160720]     chrX 154611909-154612034      * | ID=SRX190209;Name=YY..</span>
<span class="co">#&gt;   [160721]     chrX 154611909-154612034      * | ID=SRX190209;Name=YY..</span>
<span class="co">#&gt;   [160722]     chrY   21154746-21154873      * | ID=SRX190209;Name=YY..</span>
<span class="co">#&gt;                score     itemRgb               thick assigned_region  distance</span>
<span class="co">#&gt;            &lt;numeric&gt; &lt;character&gt;           &lt;IRanges&gt;       &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;        [1]       109     #006FFF       762751-762895              21    -47807</span>
<span class="co">#&gt;        [2]       109     #006FFF       762751-762895              22    -48397</span>
<span class="co">#&gt;        [3]       109     #006FFF       762751-762895              23    -48579</span>
<span class="co">#&gt;        [4]       109     #006FFF       762751-762895              24    -48579</span>
<span class="co">#&gt;        [5]       109     #006FFF       762751-762895              25    -48579</span>
<span class="co">#&gt;        ...       ...         ...                 ...             ...       ...</span>
<span class="co">#&gt;   [160718]       388     #00FF72 154611695-154611857           78347   -2110.0</span>
<span class="co">#&gt;   [160719]       388     #00FF72 154611695-154611857           78348   -2110.0</span>
<span class="co">#&gt;   [160720]        88     #0059FF 154611909-154612034           78347   -1914.5</span>
<span class="co">#&gt;   [160721]        88     #0059FF 154611909-154612034           78348   -1914.5</span>
<span class="co">#&gt;   [160722]       163     #00A6FF   21154746-21154873           78747  -49796.5</span>
<span class="co">#&gt;            peak_score</span>
<span class="co">#&gt;             &lt;numeric&gt;</span>
<span class="co">#&gt;        [1]  0.0896107</span>
<span class="co">#&gt;        [2]  0.0875207</span>
<span class="co">#&gt;        [3]  0.0868859</span>
<span class="co">#&gt;        [4]  0.0868859</span>
<span class="co">#&gt;        [5]  0.0868859</span>
<span class="co">#&gt;        ...        ...</span>
<span class="co">#&gt;   [160718]  0.5574402</span>
<span class="co">#&gt;   [160719]  0.5574402</span>
<span class="co">#&gt;   [160720]  0.5618165</span>
<span class="co">#&gt;   [160721]  0.5618165</span>
<span class="co">#&gt;   [160722]  0.0827559</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 56 sequences from an unspecified genome; no seqlengths</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $YY2</span>
<span class="co">#&gt; GRanges object with 20393 ranges and 7 metadata columns:</span>
<span class="co">#&gt;           seqnames              ranges strand |                   name</span>
<span class="co">#&gt;              &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |            &lt;character&gt;</span>
<span class="co">#&gt;       [1]     chr1       894578-894665      * | ID=SRX1530266;Name=Y..</span>
<span class="co">#&gt;       [2]     chr1       894578-894665      * | ID=SRX1530266;Name=Y..</span>
<span class="co">#&gt;       [3]     chr1       894578-894665      * | ID=SRX1530266;Name=Y..</span>
<span class="co">#&gt;       [4]     chr1       894578-894665      * | ID=SRX1530266;Name=Y..</span>
<span class="co">#&gt;       [5]     chr1       894578-894665      * | ID=SRX1530266;Name=Y..</span>
<span class="co">#&gt;       ...      ...                 ...    ... .                    ...</span>
<span class="co">#&gt;   [20389]     chrX 154299744-154299864      * | ID=SRX1530266;Name=Y..</span>
<span class="co">#&gt;   [20390]     chrX 154444690-154444806      * | ID=SRX1530266;Name=Y..</span>
<span class="co">#&gt;   [20391]     chrX 154444690-154444806      * | ID=SRX1530266;Name=Y..</span>
<span class="co">#&gt;   [20392]     chrX 154611688-154611827      * | ID=SRX1530266;Name=Y..</span>
<span class="co">#&gt;   [20393]     chrX 154611688-154611827      * | ID=SRX1530266;Name=Y..</span>
<span class="co">#&gt;               score     itemRgb               thick assigned_region  distance</span>
<span class="co">#&gt;           &lt;numeric&gt; &lt;character&gt;           &lt;IRanges&gt;       &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;       [1]       150     #0099FF       894578-894665              55   48555.5</span>
<span class="co">#&gt;       [2]       150     #0099FF       894578-894665              56   47693.5</span>
<span class="co">#&gt;       [3]       150     #0099FF       894578-894665              57   47513.5</span>
<span class="co">#&gt;       [4]       150     #0099FF       894578-894665              58   47061.5</span>
<span class="co">#&gt;       [5]       150     #0099FF       894578-894665              59   46787.5</span>
<span class="co">#&gt;       ...       ...         ...                 ...             ...       ...</span>
<span class="co">#&gt;   [20389]       141     #008FFF 154299744-154299864           78345  -49644.0</span>
<span class="co">#&gt;   [20390]       125     #007FFF 154444690-154444806           76878   49948.0</span>
<span class="co">#&gt;   [20391]       125     #007FFF 154444690-154444806           76879   49948.0</span>
<span class="co">#&gt;   [20392]       192     #00C3FF 154611688-154611827           78347   -2128.5</span>
<span class="co">#&gt;   [20393]       192     #00C3FF 154611688-154611827           78348   -2128.5</span>
<span class="co">#&gt;           peak_score</span>
<span class="co">#&gt;            &lt;numeric&gt;</span>
<span class="co">#&gt;       [1]  0.0869676</span>
<span class="co">#&gt;       [2]  0.0900185</span>
<span class="co">#&gt;       [3]  0.0906690</span>
<span class="co">#&gt;       [4]  0.0923232</span>
<span class="co">#&gt;       [5]  0.0933406</span>
<span class="co">#&gt;       ...        ...</span>
<span class="co">#&gt;   [20389]  0.0832623</span>
<span class="co">#&gt;   [20390]  0.0822559</span>
<span class="co">#&gt;   [20391]  0.0822559</span>
<span class="co">#&gt;   [20392]  0.5570279</span>
<span class="co">#&gt;   [20393]  0.5570279</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 41 sequences from an unspecified genome; no seqlengths</span>

<span class="co"># show added columns in the output</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">peaks</span><span class="op">$</span><span class="va">YY1</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] "name"    "score"   "itemRgb" "thick"</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">ap</span><span class="op">$</span><span class="va">YY1</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] "name"            "score"           "itemRgb"         "thick"          </span>
<span class="co">#&gt; [5] "assigned_region" "distance"        "peak_score"</span>

<span class="co"># show information in the added columns</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">ap</span><span class="op">$</span><span class="va">YY1</span><span class="op">$</span><span class="va">assigned_region</span><span class="op">)</span>
<span class="co">#&gt; [1] 21 22 23 24 25 26</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">ap</span><span class="op">$</span><span class="va">YY1</span><span class="op">$</span><span class="va">distance</span><span class="op">)</span>
<span class="co">#&gt; [1] -47807 -48397 -48579 -48579 -48579 -48579</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">ap</span><span class="op">$</span><span class="va">YY1</span><span class="op">$</span><span class="va">peak_score</span><span class="op">)</span>
<span class="co">#&gt; [1] 0.08961075 0.08752069 0.08688586 0.08688586 0.08688586 0.08688586</span></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># get direct targets</span>
<span class="va">dt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map2.html">map2</a></span><span class="op">(</span><span class="va">peaks</span>, <span class="va">regions</span>,
           <span class="op">~</span><span class="fu"><a href="https://rdrr.io/pkg/target/man/direct_targets.html">direct_targets</a></span><span class="op">(</span>peaks<span class="op">=</span><span class="va">.x</span>,
                           regions <span class="op">=</span> <span class="va">.y</span>,
                           regions_col <span class="op">=</span> <span class="st">'tx_id'</span>,
                           stats_col <span class="op">=</span> <span class="st">'fc'</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># show direct_targets return</span>
<span class="va">dt</span>
<span class="co">#&gt; $YY1</span>
<span class="co">#&gt; GRanges object with 44403 ranges and 13 metadata columns:</span>
<span class="co">#&gt;                seqnames        ranges strand |     tx_id     tx_name gene_id</span>
<span class="co">#&gt;                   &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;  &lt;list&gt;</span>
<span class="co">#&gt;        1           chr1 760530-860729      + |        21  uc001abv.1  148398</span>
<span class="co">#&gt;        6           chr1 761121-861320      + |        22  uc001abw.1  148398</span>
<span class="co">#&gt;       11           chr1 761302-861501      + |        23  uc031pjl.1  148398</span>
<span class="co">#&gt;       16           chr1 761302-861501      + |        24  uc031pjm.1  148398</span>
<span class="co">#&gt;       21           chr1 761302-861501      + |        25  uc031pjn.1  148398</span>
<span class="co">#&gt;      ...            ...           ...    ... .       ...         ...     ...</span>
<span class="co">#&gt;   160718 chrUn_gl000223  68588-168787      - |     82934  uc011mgb.1    7637</span>
<span class="co">#&gt;   160719 chrUn_gl000223  68588-168787      - |     82935  uc011mgc.1    7637</span>
<span class="co">#&gt;   160720 chrUn_gl000223 119531-219730      - |     82936  uc011mgd.2    7574</span>
<span class="co">#&gt;   160721 chrUn_gl000223 119531-219730      - |     82937  uc031tgg.1    7574</span>
<span class="co">#&gt;   160722 chrUn_gl000223 119531-219730      - |     82938  uc031tgh.1    7574</span>
<span class="co">#&gt;                 gene          tf      gene.1        fc    pvalue     score</span>
<span class="co">#&gt;          &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;        1      SAMD11         YY1      SAMD11  -0.54365   0.05581  0.953844</span>
<span class="co">#&gt;        6      SAMD11         YY1      SAMD11  -0.54365   0.05581  0.944972</span>
<span class="co">#&gt;       11      SAMD11         YY1      SAMD11  -0.54365   0.05581  0.942342</span>
<span class="co">#&gt;       16      SAMD11         YY1      SAMD11  -0.54365   0.05581  0.942342</span>
<span class="co">#&gt;       21      SAMD11         YY1      SAMD11  -0.54365   0.05581  0.942342</span>
<span class="co">#&gt;      ...         ...         ...         ...       ...       ...       ...</span>
<span class="co">#&gt;   160718       ZNF84         YY1       ZNF84   0.08999   0.68511  0.175405</span>
<span class="co">#&gt;   160719       ZNF84         YY1       ZNF84   0.08999   0.68511  0.175405</span>
<span class="co">#&gt;   160720       ZNF26         YY1       ZNF26  -0.47400   0.09301  0.273345</span>
<span class="co">#&gt;   160721       ZNF26         YY1       ZNF26  -0.47400   0.09301  0.273345</span>
<span class="co">#&gt;   160722       ZNF26         YY1       ZNF26  -0.47400   0.09301  0.273345</span>
<span class="co">#&gt;          score_rank      stat stat_rank      rank</span>
<span class="co">#&gt;           &lt;integer&gt; &lt;numeric&gt; &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;        1      14440  -0.54365      7623 0.0558301</span>
<span class="co">#&gt;        6      14621  -0.54365      7623 0.0565299</span>
<span class="co">#&gt;       11      14671  -0.54365      7623 0.0567232</span>
<span class="co">#&gt;       16      14671  -0.54365      7623 0.0567232</span>
<span class="co">#&gt;       21      14671  -0.54365      7623 0.0567232</span>
<span class="co">#&gt;      ...        ...       ...       ...       ...</span>
<span class="co">#&gt;   160718      37285   0.08999     35329  0.668099</span>
<span class="co">#&gt;   160719      37285   0.08999     35329  0.668099</span>
<span class="co">#&gt;   160720      33771  -0.47400      9775  0.167431</span>
<span class="co">#&gt;   160721      33771  -0.47400      9775  0.167431</span>
<span class="co">#&gt;   160722      33771  -0.47400      9775  0.167431</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 93 sequences from an unspecified genome; no seqlengths</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $YY2</span>
<span class="co">#&gt; GRanges object with 15296 ranges and 13 metadata columns:</span>
<span class="co">#&gt;         seqnames              ranges strand |     tx_id     tx_name gene_id</span>
<span class="co">#&gt;            &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;  &lt;list&gt;</span>
<span class="co">#&gt;       1     chr1       795967-896166      + |        55  uc001aca.2  339451</span>
<span class="co">#&gt;       2     chr1       796829-897028      + |        56  uc001acb.1  339451</span>
<span class="co">#&gt;       3     chr1       797009-897208      + |        57  uc010nya.1  339451</span>
<span class="co">#&gt;       4     chr1       797461-897660      + |        58  uc001acc.2  339451</span>
<span class="co">#&gt;       5     chr1       797735-897934      + |        59  uc010nyb.1  339451</span>
<span class="co">#&gt;     ...      ...                 ...    ... .       ...         ...     ...</span>
<span class="co">#&gt;   20388     chrX 154134649-154234848      - |     78342  uc010nvi.1    2157</span>
<span class="co">#&gt;   20389     chrX 154255016-154355215      - |     78343  uc011mzx.1    2157</span>
<span class="co">#&gt;   20391     chrX 154299348-154399547      - |     78345  uc004fmz.2    4515</span>
<span class="co">#&gt;   20392     chrX 154563787-154663986      - |     78347  uc004fnf.3    1193</span>
<span class="co">#&gt;   20393     chrX 154563787-154663986      - |     78348  uc010nvj.1    1193</span>
<span class="co">#&gt;                gene          tf      gene.1        fc    pvalue     score</span>
<span class="co">#&gt;         &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;       1      KLHL17         YY2      KLHL17  -0.39191   0.30656 0.0869676</span>
<span class="co">#&gt;       2      KLHL17         YY2      KLHL17  -0.39191   0.30656 0.0900185</span>
<span class="co">#&gt;       3      KLHL17         YY2      KLHL17  -0.39191   0.30656 0.0906690</span>
<span class="co">#&gt;       4      KLHL17         YY2      KLHL17  -0.39191   0.30656 0.0923232</span>
<span class="co">#&gt;       5      KLHL17         YY2      KLHL17  -0.39191   0.30656 0.0933406</span>
<span class="co">#&gt;     ...         ...         ...         ...       ...       ...       ...</span>
<span class="co">#&gt;   20388          F8         YY2          F8   0.32501   0.12452 0.3079124</span>
<span class="co">#&gt;   20389          F8         YY2          F8   0.32501   0.12452 0.5732370</span>
<span class="co">#&gt;   20391       MTCP1         YY2       MTCP1  -0.06521   0.65344 0.0832623</span>
<span class="co">#&gt;   20392       CLIC2         YY2       CLIC2   0.21019   0.35482 0.5570279</span>
<span class="co">#&gt;   20393       CLIC2         YY2       CLIC2   0.21019   0.35482 0.5570279</span>
<span class="co">#&gt;         score_rank      stat stat_rank      rank</span>
<span class="co">#&gt;          &lt;integer&gt; &lt;numeric&gt; &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;       1      11139  -0.39191      2527  0.120308</span>
<span class="co">#&gt;       2      11002  -0.39191      2527  0.118829</span>
<span class="co">#&gt;       3      10973  -0.39191      2527  0.118515</span>
<span class="co">#&gt;       4      10902  -0.39191      2527  0.117749</span>
<span class="co">#&gt;       5      10865  -0.39191      2527  0.117349</span>
<span class="co">#&gt;     ...        ...       ...       ...       ...</span>
<span class="co">#&gt;   20388       5339   0.32501      3699 0.0844090</span>
<span class="co">#&gt;   20389       1707   0.32501      3699 0.0269875</span>
<span class="co">#&gt;   20391      11837  -0.06521     12032 0.6087286</span>
<span class="co">#&gt;   20392       1908   0.21019      6592 0.0537576</span>
<span class="co">#&gt;   20393       1908   0.21019      6592 0.0537576</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 93 sequences from an unspecified genome; no seqlengths</span>

<span class="co"># show added columns in the output</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">regions</span><span class="op">$</span><span class="va">YY1</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] "tx_id"   "tx_name" "gene_id" "gene"    "tf"      "gene"    "fc"     </span>
<span class="co">#&gt; [8] "pvalue"</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">dt</span><span class="op">$</span><span class="va">YY1</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt;  [1] "tx_id"      "tx_name"    "gene_id"    "gene"       "tf"        </span>
<span class="co">#&gt;  [6] "gene.1"     "fc"         "pvalue"     "score"      "score_rank"</span>
<span class="co">#&gt; [11] "stat"       "stat_rank"  "rank"</span>

<span class="co"># show information in the added columns</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">dt</span><span class="op">$</span><span class="va">YY1</span><span class="op">$</span><span class="va">score</span><span class="op">)</span>
<span class="co">#&gt; [1] 0.9538438 0.9449725 0.9423423 0.9423423 0.9423423 0.9423423</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">dt</span><span class="op">$</span><span class="va">YY1</span><span class="op">$</span><span class="va">score_rank</span><span class="op">)</span>
<span class="co">#&gt; [1] 14440 14621 14671 14671 14671 14671</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">dt</span><span class="op">$</span><span class="va">YY1</span><span class="op">$</span><span class="va">rank</span><span class="op">)</span>
<span class="co">#&gt; [1] 0.05583011 0.05652992 0.05672324 0.05672324 0.05672324 0.05672324</span></code></pre></div>
<p>To determine the dominant function of a factor,</p>
<ul>
<li>We divide the targets into groups based on the effect of the knockdown</li>
<li>We use the empirical distribution function (ECDF) to show the fraction of targets at a specified regulatory potential value or less.</li>
</ul>
<p>Because the ranks rather than the absolute value of the regulatory potential is used, the lower the value the higher the potential. Then the groups of targets can be compared to each other or to a theoretical distribution.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Figure 2</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span>

<span class="co"># plot distance by score of associate peaks</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">ap</span><span class="op">$</span><span class="va">YY1</span><span class="op">$</span><span class="va">distance</span>, <span class="va">ap</span><span class="op">$</span><span class="va">YY1</span><span class="op">$</span><span class="va">peak_score</span>,
     xlab <span class="op">=</span> <span class="st">'Distance'</span>, ylab <span class="op">=</span> <span class="st">'Peak Score'</span>,
     main <span class="op">=</span> <span class="st">'(A)'</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/points.html">points</a></span><span class="op">(</span><span class="va">ap</span><span class="op">$</span><span class="va">YY2</span><span class="op">$</span><span class="va">distance</span>, <span class="va">ap</span><span class="op">$</span><span class="va">YY2</span><span class="op">$</span><span class="va">peak_score</span><span class="op">)</span>

<span class="co"># make labels, colors and groups</span>
<span class="va">labs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'Down'</span>, <span class="st">'None'</span>, <span class="st">'Up'</span><span class="op">)</span>
<span class="va">cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'green'</span>, <span class="st">'gray'</span>, <span class="st">'red'</span><span class="op">)</span>

<span class="co"># make three groups by quantiles      </span>
<span class="va">groups</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map</a></span><span class="op">(</span><span class="va">dt</span>,<span class="op">~</span><span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/base/cut.html">cut</a></span><span class="op">(</span><span class="va">.x</span><span class="op">$</span><span class="va">stat</span>, breaks <span class="op">=</span> <span class="fl">3</span>, labels <span class="op">=</span> <span class="va">labs</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span>

<span class="co"># plot the group functions</span>
<span class="fu"><a href="https://purrr.tidyverse.org/reference/map2.html">pmap</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">dt</span>, <span class="va">groups</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'(B)'</span>, <span class="st">'(C)'</span><span class="op">)</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">y</span>, <span class="va">z</span><span class="op">)</span> <span class="op">{</span>
      <span class="fu"><a href="https://rdrr.io/pkg/target/man/plot_predictions.html">plot_predictions</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">score_rank</span>,
                       group <span class="op">=</span> <span class="va">y</span>, colors <span class="op">=</span> <span class="va">cols</span>, labels <span class="op">=</span> <span class="va">labs</span>,
                       xlab <span class="op">=</span> <span class="st">'Regulatory Potential'</span>, ylab <span class="op">=</span> <span class="st">'ECDF'</span><span class="op">)</span>
      <span class="fu"><a href="https://rdrr.io/r/graphics/title.html">title</a></span><span class="op">(</span><span class="va">z</span><span class="op">)</span>
    <span class="op">}</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="workshop_code_files/figure-html/functions-1.png" alt="Figure 2. Predicted functions of YY1 and YY2 on their specific targets. Bindings peaks of the transcription factors in HeLa cells were determined using ChIP-Seq. Distances from the transcription start sites and the transformed distances of the (A) YY1 and YY2 peaks are shown. The regulatory potential of each gene was calculated using target. Genes were grouped into up, none or down regulated based on the fold-change. The emperical cumulative distribution functions (ECDF) of the groups of (C) YY1 and (D) YY2 targets are shown at each regulatory potential value." width="100%"><p class="caption">
Figure 2. Predicted functions of YY1 and YY2 on their specific targets. Bindings peaks of the transcription factors in HeLa cells were determined using ChIP-Seq. Distances from the transcription start sites and the transformed distances of the (A) YY1 and YY2 peaks are shown. The regulatory potential of each gene was calculated using target. Genes were grouped into up, none or down regulated based on the fold-change. The emperical cumulative distribution functions (ECDF) of the groups of (C) YY1 and (D) YY2 targets are shown at each regulatory potential value.
</p>
</div>
<p>The scores of the individual peaks are a decreasing function of the distance from the transcription start sites.</p>
<p>The closer the factor binding site from the start site the lower the score. The distribution of these scores is very similar for both factors (Figure 2A). The ECDF of the down-regulated targets of YY1 is higher than that of up- and none-regulated targets (Figure 2B). Therefore, the absence of YY1 on its targets result in aggregate in their down regulation.</p>
<p>The opposite is true for YY2 where more high ranking targets are up-regulated by the factor knockdown (Figure 2C).</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Table 2</span>
<span class="co"># test individual factor functions</span>
<span class="fu"><a href="https://purrr.tidyverse.org/reference/map2.html">map2</a></span><span class="op">(</span><span class="va">dt</span>, <span class="va">groups</span>,
     <span class="op">~</span><span class="fu"><a href="https://rdrr.io/pkg/target/man/test_predictions.html">test_predictions</a></span><span class="op">(</span><span class="va">.x</span><span class="op">$</span><span class="va">rank</span>,
                       group <span class="op">=</span> <span class="va">.y</span>,
                       compare <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'Down'</span>, <span class="st">'Up'</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; $YY1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;  Two-sample Kolmogorov-Smirnov test</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; data:  x and y</span>
<span class="co">#&gt; D = 0.78597, p-value &lt; 2.2e-16</span>
<span class="co">#&gt; alternative hypothesis: two-sided</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $YY2</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;  Two-sample Kolmogorov-Smirnov test</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; data:  x and y</span>
<span class="co">#&gt; D = 0.40354, p-value = 1.075e-12</span>
<span class="co">#&gt; alternative hypothesis: two-sided</span></code></pre></div>
<p>To formally test these observations, we use the Kolmogorov-Smirnov (KS) test. The distribution of the two groups are compared for equality. If one lies one either side of the other then they must be drawn from different distributions.</p>
<p>Here, we compared the up- and down-regulated functions for both factors (Table 2).</p>
<p>In both cases, the distribution of the two groups were significantly different from one another.</p>
</div>
<div id="predicting-the-shared-targets-of-the-two-factors" class="section level1">
<h1 class="hasAnchor">
<a href="#predicting-the-shared-targets-of-the-two-factors" class="anchor"></a>Predicting the shared targets of the two factors</h1>
<p>Similar to the previous analysis,</p>
<ul>
<li>Identify shared/common peaks using <code>subsetByOverlaps</code>.</li>
<li>Pass two columns to <code>stats_col</code> to use the fold-changes from both factors</li>
<li>
<code>common_peaks</code> and <code>both_regions</code> are the main inputs</li>
</ul>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># merge and name peaks</span>
<span class="va">common_peaks</span> <span class="op">&lt;-</span> <span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/inter-range-methods.html">reduce</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html">subsetByOverlaps</a></span><span class="op">(</span><span class="va">peaks</span><span class="op">$</span><span class="va">YY1</span>, <span class="va">peaks</span><span class="op">$</span><span class="va">YY2</span><span class="op">)</span><span class="op">)</span>
<span class="va">common_peaks</span><span class="op">$</span><span class="va">name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">'common_peak_'</span>, <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">common_peaks</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># bind express tables into one</span>
<span class="va">both_express</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html">bind_rows</a></span><span class="op">(</span><span class="va">express</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_wider.html">pivot_wider</a></span><span class="op">(</span>names_from <span class="op">=</span> <span class="va">tf</span>,
              values_from <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">fc</span>, <span class="va">pvalue</span><span class="op">)</span><span class="op">)</span>

<span class="co"># match symbols between genome and expression objects</span>
<span class="va">ind</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span><span class="op">(</span><span class="va">genome</span><span class="op">$</span><span class="va">gene</span>, <span class="va">both_express</span><span class="op">$</span><span class="va">gene</span><span class="op">)</span>

<span class="co"># subset genome by genes with expression info</span>
<span class="va">both_regions</span> <span class="op">&lt;-</span> <span class="va">genome</span><span class="op">[</span><span class="va">genome</span><span class="op">$</span><span class="va">gene</span> <span class="op">%in%</span> <span class="va">both_express</span><span class="op">$</span><span class="va">gene</span><span class="op">]</span>

<span class="co"># add expression info to gene info</span>
<span class="fu">mcols</span><span class="op">(</span><span class="va">both_regions</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">both_regions</span><span class="op">)</span>, <span class="va">both_express</span><span class="op">[</span><span class="va">ind</span>,<span class="op">]</span><span class="op">)</span>  </code></pre></div>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># get associated peaks with both factors</span>
<span class="va">common_ap</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/target/man/associated_peaks.html">associated_peaks</a></span><span class="op">(</span>peaks <span class="op">=</span> <span class="va">common_peaks</span>,
                              regions <span class="op">=</span> <span class="va">both_regions</span>,
                              regions_col <span class="op">=</span> <span class="st">'tx_id'</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># get direct targets of both factors</span>
<span class="va">common_dt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/target/man/direct_targets.html">direct_targets</a></span><span class="op">(</span>peaks <span class="op">=</span> <span class="va">common_peaks</span>,
                            regions <span class="op">=</span> <span class="va">both_regions</span>,
                            regions_col <span class="op">=</span> <span class="st">'tx_id'</span>,
                            stats_col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'fc_YY1'</span>, <span class="st">'fc_YY2'</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<ul>
<li>
<code>associated_peaks</code> is the same as before.</li>
<li>
<code>direct_targets</code> is the same but the <code>stat</code> and the <code>stat_rank</code> carry the product of the two statistics provided in the previous step and the rank of that product.</li>
</ul>
<p>The output can also be visualized the same way. The targets are divided into three groups based on the statistics product.</p>
<ul>
<li>When the two statistics agree in the sign, the product is positive. This means the knockdown of either transcription factor results in same direction change in the target gene expression. Therefore, the two factors would cooperate if they bind to the same site on that gene.</li>
<li>The reverse is true for targets with opposite signed statistics. On these targets, the two factors would be expected to compete for inducing opposing changes in the expression.</li>
</ul>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Figure 3</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span>
<span class="co"># plot distiace by score for associated peaks</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">common_ap</span><span class="op">$</span><span class="va">distance</span>,
     <span class="va">common_ap</span><span class="op">$</span><span class="va">peak_score</span>,
     xlab <span class="op">=</span> <span class="st">'Distance'</span>,
     ylab <span class="op">=</span> <span class="st">'Peak Score'</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/title.html">title</a></span><span class="op">(</span><span class="st">'(A)'</span><span class="op">)</span>

<span class="co"># make labels, colors and gorups</span>
<span class="va">labs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'Competitive'</span>, <span class="st">'None'</span>, <span class="st">'Cooperative'</span><span class="op">)</span>
<span class="va">cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'green'</span>, <span class="st">'gray'</span>, <span class="st">'red'</span><span class="op">)</span>

<span class="co"># make three groups by quantiles      </span>
<span class="va">common_groups</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cut.html">cut</a></span><span class="op">(</span><span class="va">common_dt</span><span class="op">$</span><span class="va">stat</span>,
                     breaks <span class="op">=</span> <span class="fl">3</span>,
                     labels <span class="op">=</span> <span class="va">labs</span><span class="op">)</span>

<span class="co"># plot predicted function</span>
<span class="fu"><a href="https://rdrr.io/pkg/target/man/plot_predictions.html">plot_predictions</a></span><span class="op">(</span><span class="va">common_dt</span><span class="op">$</span><span class="va">score_rank</span>,
                 group <span class="op">=</span> <span class="va">common_groups</span>,
                 colors <span class="op">=</span> <span class="va">cols</span>, labels <span class="op">=</span> <span class="va">labs</span>,
                 xlab <span class="op">=</span> <span class="st">'Regulatory Interaction'</span>, ylab <span class="op">=</span> <span class="st">'ECDF'</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/title.html">title</a></span><span class="op">(</span><span class="st">'(B)'</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="workshop_code_files/figure-html/function-1.png" alt="Figure 3. Predicted function of YY1 and YY2 on their shared targets. Shared bindings sites of YY1 and YY2 in HeLa cells were determined using the overlap of the individual factor ChIP-Seq peaks. (A) Distances from the transcription start sites and the transformed distances of the shared peaks are shown. The regulatory interaction of each gene was calculated using target. Genes were grouped into cooperatively, none or competitively regulated based on the the product of the fold-changes from YY1- and YY2-knockdown. (B) The emperical cumulative distribution functions (ECDF) of the groups of targets are shown at each regulatory potential value." width="100%"><p class="caption">
Figure 3. Predicted function of YY1 and YY2 on their shared targets. Shared bindings sites of YY1 and YY2 in HeLa cells were determined using the overlap of the individual factor ChIP-Seq peaks. (A) Distances from the transcription start sites and the transformed distances of the shared peaks are shown. The regulatory interaction of each gene was calculated using target. Genes were grouped into cooperatively, none or competitively regulated based on the the product of the fold-changes from YY1- and YY2-knockdown. (B) The emperical cumulative distribution functions (ECDF) of the groups of targets are shown at each regulatory potential value.
</p>
</div>
<ul>
<li>The common peaks distances and scores take the same shape (Figure 3A).</li>
<li>The two factors seem to cooperate on more of the common target than any of the two other possibilities (Figure 3B).</li>
</ul>
<p>This observation can be tested using the KS test. The curve of the cooperative targets lies above that of none and competitively regulated targets (Table 3).</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Table 3</span>
<span class="co"># test factors are cooperative</span>
<span class="fu"><a href="https://rdrr.io/pkg/target/man/test_predictions.html">test_predictions</a></span><span class="op">(</span><span class="va">common_dt</span><span class="op">$</span><span class="va">score_rank</span>,
                 group <span class="op">=</span> <span class="va">common_groups</span>,
                 compare <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'Cooperative'</span>, <span class="st">'None'</span><span class="op">)</span>,
                 alternative <span class="op">=</span> <span class="st">'greater'</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;  Two-sample Kolmogorov-Smirnov test</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; data:  x and y</span>
<span class="co">#&gt; D^+ = 0.30253, p-value = 0.01034</span>
<span class="co">#&gt; alternative hypothesis: the CDF of x lies above that of y</span>

<span class="co"># test factors are more cooperative than competitive</span>
<span class="fu"><a href="https://rdrr.io/pkg/target/man/test_predictions.html">test_predictions</a></span><span class="op">(</span><span class="va">common_dt</span><span class="op">$</span><span class="va">score_rank</span>,
                 group <span class="op">=</span> <span class="va">common_groups</span>,
                 compare <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'Cooperative'</span>, <span class="st">'Competitive'</span><span class="op">)</span>,
                 alternative <span class="op">=</span> <span class="st">'greater'</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;  Two-sample Kolmogorov-Smirnov test</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; data:  x and y</span>
<span class="co">#&gt; D^+ = 0.29706, p-value = 0.01263</span>
<span class="co">#&gt; alternative hypothesis: the CDF of x lies above that of y</span></code></pre></div>
</div>
<div id="summary" class="section level1">
<h1 class="hasAnchor">
<a href="#summary" class="anchor"></a>Summary</h1>
<ul>
<li>I presented a workflow for predicting the direct targets of a transcription factor by integrating binding and expression data.</li>
<li>The <code>target</code> package implements the BETA algorithm to rank targets based on the distance of the ChIP peaks of the transcription factor in the genes and the differential expression from the factor perturbation.</li>
<li>To predict the combined function of two factors, two sets of data are used to find the shared peaks and the product of their differential expression.</li>
</ul>
<div id="open-discussion-next" class="section level3">
<h3 class="hasAnchor">
<a href="#open-discussion-next" class="anchor"></a>Open discussion (–&gt; Next)</h3>
</div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Chen2010" class="csl-entry">
Chen, Li, Toshi Shioda, Kathryn R. Coser, Mary C. Lynch, Chuanwei Yang, and Emmett V. Schmidt. 2010. <span>“<span class="nocase">Genome-wide analysis of YY2 versus YY1 target genes</span>.”</span> <em>Nucleic Acids Research</em> 38 (12): 4011–26. <a href="https://doi.org/10.1093/nar/gkq112">https://doi.org/10.1093/nar/gkq112</a>.
</div>
<div id="ref-Michaud2013" class="csl-entry">
Michaud, Joëlle, Viviane Praz, Nicole James Faresse, Courtney K Jnbaptiste, Shweta Tyagi, Frédéric Schütz, and Winship Herr. 2013. <span>“<span class="nocase">HCFC1 is a common component of active human CpG-island promoters and coincides with ZNF143, THAP11, YY1, and GABP transcription factor occupancy.</span>”</span> <em>Genome Research</em> 23 (6): 907–16. <a href="https://doi.org/10.1101/gr.150078.112">https://doi.org/10.1101/gr.150078.112</a>.
</div>
<div id="ref-Wu2017d" class="csl-entry">
Wu, Xiao-Nan, Tao-Tao Shi, Yao-Hui He, Fei-Fei Wang, Rui Sang, Jian-Cheng Ding, Wen-Juan Zhang, et al. 2017. <span>“<span class="nocase">Methylation of transcription factor YY2 regulates its transcriptional activity and cell proliferation.</span>”</span> <em>Cell Discovery</em> 3: 17035. <a href="https://doi.org/10.1038/celldisc.2017.35">https://doi.org/10.1038/celldisc.2017.35</a>.
</div>
</div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p><a href="mailto:mahshaaban@gnu.ac.kr" class="email">mahshaaban@gnu.ac.kr</a><a href="#fnref1" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Mahmoud Ahmed.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
