<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction: Introduction to target package • IntegratingDataBioc</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction: Introduction to target package">
<meta property="og:description" content="IntegratingDataBioc">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-93043521-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-93043521-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">IntegratingDataBioc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/workshop_code.html">Code Walkthrough: A use case of YY1 and YY2 in HeLa cells</a>
    </li>
    <li>
      <a href="../articles/workshop_introduction.html">Introduction: Introduction to target package</a>
    </li>
    <li>
      <a href="../articles/workshop_packages.html">Introduction Bioconductor packages</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/MahShaaban/IntegratingDataBioc">
    <span class="fas fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="workshop_introduction_files/header-attrs-2.6/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction: Introduction to target package</h1>
                        <h4 class="author">Mahmoud Ahmed<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>
</h4>
            
            <h4 class="date">2021-01-23</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/MahShaaban/IntegratingDataBioc/blob/master/vignettes/workshop_introduction.Rmd"><code>vignettes/workshop_introduction.Rmd</code></a></small>
      <div class="hidden name"><code>workshop_introduction.Rmd</code></div>

    </div>

    
    
<div id="background" class="section level1">
<h1 class="hasAnchor">
<a href="#background" class="anchor"></a>Background</h1>
<div id="predicting-the-interaction-of-two-factors-using-binding-and-expression-data" class="section level2">
<h2 class="hasAnchor">
<a href="#predicting-the-interaction-of-two-factors-using-binding-and-expression-data" class="anchor"></a>Predicting the interaction of two factors using binding and expression data</h2>
<ul>
<li><p>The binding of a transcription factor to a regulatory region (e.g. gene promoter) induces or represses its expression <span class="citation">Rougemont and Naef (2012)</span>.</p></li>
<li><p>Transcription factors share their binding sites with other factor, co-factors and/or DNA-binding proteins forming complexes.</p></li>
<li><p>The integration of the overlapping binding sites and the effect on gene expression of perturbing the factors can be used to infer their combined function; cooperative or competitive.</p></li>
</ul>
</div>
<div id="modeling-the-binding-sites-as-the-discounted-distances-of-the-chip-peaks" class="section level2">
<h2 class="hasAnchor">
<a href="#modeling-the-binding-sites-as-the-discounted-distances-of-the-chip-peaks" class="anchor"></a>Modeling the binding sites as the discounted distances of the ChIP peaks</h2>
<p>Peak Score (<span class="math inline">\(S_p\)</span>): is the distance (<span class="math inline">\(\Delta\)</span>) from transcription start site (TSS) relative to a 100 kb <span class="citation">Wang et al. (2013)</span>.</p>
<p><span class="math display">\[
S_p = e^{-(0.5+4\Delta)}
\]</span></p>
<p>The shape of the function approximate empirical observations <span class="citation">Tang et al. (2011)</span></p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># define distances</span>
<span class="va">d</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="fl">100</span>

<span class="co"># define a decay function</span>
<span class="va">decay_func</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">exp</a></span><span class="op">(</span><span class="op">-</span><span class="op">(</span><span class="fl">0.5</span> <span class="op">+</span> <span class="op">(</span><span class="fl">4</span> <span class="op">*</span> <span class="va">x</span><span class="op">/</span><span class="fl">100</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="co"># plot decay over distances</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">d</span>, <span class="fu">decay_func</span><span class="op">(</span><span class="va">d</span><span class="op">)</span>,
     pch <span class="op">=</span> <span class="fl">19</span>, 
     xlab <span class="op">=</span> <span class="st">'d = 1:100'</span>, ylab <span class="op">=</span> <span class="st">'exp(-(0.5 + 4 * d/100))'</span><span class="op">)</span></code></pre></div>
<p><img src="workshop_introduction_files/figure-html/decay_function-1.png" width="480" style="display: block; margin: auto;"></p>
</div>
<div id="modeling-the-regulatory-potential-as-the-sum-of-the-weighted-peaks" class="section level2">
<h2 class="hasAnchor">
<a href="#modeling-the-regulatory-potential-as-the-sum-of-the-weighted-peaks" class="anchor"></a>Modeling the regulatory potential as the sum of the weighted peaks</h2>
<p>Gene Score (<span class="math inline">\(S_g\)</span>): is the sum of the scores (<span class="math inline">\(S_p\)</span>) of the <span class="math inline">\(k\)</span> peaks nearby the TSS.</p>
<p><span class="math display">\[S_g = \sum_{i=1}^k S_{pi}\]</span></p>
<p>Regulatory potential increases with the number of binding sites <span class="citation">Tang et al. (2011)</span>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># define distances</span>
<span class="va">d</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="fl">100</span>

<span class="co"># sample distances to assign genes</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span>
<span class="va">g</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-envirAPI.html">sample</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="fl">100</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="co"># count the number of instances</span>
<span class="va">n</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">)</span>

<span class="co"># aggregate the distances for each instance</span>
<span class="va">s</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/aggregate.html">aggregate</a></span><span class="op">(</span><span class="va">d</span>, <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span>, <span class="va">sum</span><span class="op">)</span><span class="op">$</span><span class="va">x</span>

<span class="co"># plot aggregate distances over the number of instances</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">n</span>, <span class="va">s</span>,
     pch <span class="op">=</span> <span class="fl">19</span>,
     xlab <span class="op">=</span> <span class="st">'table(g)'</span>, ylab <span class="op">=</span> <span class="st">'aggregate(d, g, sum)'</span><span class="op">)</span></code></pre></div>
<p><img src="workshop_introduction_files/figure-html/aggregate_functions-1.png" width="480" style="display: block; margin: auto;"></p>
</div>
<div id="integrating-the-factor-binding-and-the-expression-information" class="section level2">
<h2 class="hasAnchor">
<a href="#integrating-the-factor-binding-and-the-expression-information" class="anchor"></a>Integrating the factor binding and the expression information</h2>
<p>Rank Product (<span class="math inline">\(RP_g\)</span>): the gene score (<span class="math inline">\(R_{gb}\)</span>) rank is multiplied by the gene statistics rank (<span class="math inline">\(R_{ge}\)</span>) from differential expression <span class="citation">Breitling et al. (2004)</span>.</p>
<p><span class="math display">\[
RP_g = \frac{R_{gb}\times R_{ge}}{n^2}
\]</span></p>
<p>Integrate the binding events and the functional effect <span class="citation">Tang et al. (2011)</span>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># define distances</span>
<span class="va">d</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="fl">100</span>

<span class="co"># make a random stat variable</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">12345</span><span class="op">)</span>
<span class="va">t</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/Normal.html">rnorm</a></span><span class="op">(</span><span class="fl">200</span>, <span class="fl">0</span>, <span class="fl">5</span><span class="op">)</span>

<span class="co"># calculate rank product</span>
<span class="va">rp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rank.html">rank</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="va">d</span>, <span class="va">d</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/base/rank.html">rank</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">t</span><span class="op">)</span><span class="op">)</span>

<span class="co"># get min and max ranks</span>
<span class="va">pmax</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">rp</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">rp</span><span class="op">)</span><span class="op">)</span>
<span class="va">pmin</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">rp</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">rp</span><span class="op">)</span><span class="op">)</span>

<span class="co"># plot random stats over distances</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="va">d</span>, <span class="va">d</span><span class="op">)</span>, <span class="va">t</span>,
     pch <span class="op">=</span> <span class="fl">19</span>, col <span class="op">=</span> <span class="st">'gray'</span>,
     xlab <span class="op">=</span> <span class="st">'c(-d, d)'</span>, ylab <span class="op">=</span> <span class="st">'rnorm(200, 0, 5)'</span><span class="op">)</span>

<span class="co"># highlight the max rank</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/points.html">points</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="va">d</span>, <span class="va">d</span><span class="op">)</span><span class="op">[</span><span class="va">pmax</span><span class="op">]</span>, <span class="va">t</span><span class="op">[</span><span class="va">pmax</span><span class="op">]</span>,
       col <span class="op">=</span> <span class="st">'blue'</span>, pch <span class="op">=</span> <span class="fl">19</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/text.html">text</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="va">d</span>, <span class="va">d</span><span class="op">)</span><span class="op">[</span><span class="va">pmax</span><span class="op">]</span><span class="op">+</span><span class="fl">10</span>, <span class="va">t</span><span class="op">[</span><span class="va">pmax</span><span class="op">]</span> <span class="op">+</span> <span class="fl">1</span>,
       col <span class="op">=</span> <span class="st">'blue'</span>, labels <span class="op">=</span> <span class="st">'Max'</span><span class="op">)</span>

<span class="co"># highlight the min rank</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/points.html">points</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="va">d</span>, <span class="va">d</span><span class="op">)</span><span class="op">[</span><span class="va">pmin</span><span class="op">]</span>, <span class="va">t</span><span class="op">[</span><span class="va">pmin</span><span class="op">]</span>,
       col <span class="op">=</span> <span class="st">'green'</span>, pch <span class="op">=</span> <span class="fl">19</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/text.html">text</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="va">d</span>, <span class="va">d</span><span class="op">)</span><span class="op">[</span><span class="va">pmin</span><span class="op">]</span> <span class="op">+</span> <span class="fl">10</span>, <span class="va">t</span><span class="op">[</span><span class="va">pmin</span><span class="op">]</span> <span class="op">+</span> <span class="fl">1</span>,
       col <span class="op">=</span> <span class="st">'green'</span>, label <span class="op">=</span> <span class="st">'Min'</span><span class="op">)</span>

<span class="co"># make quadrant lines</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span><span class="op">(</span>h <span class="op">=</span> <span class="fl">0</span>, col <span class="op">=</span> <span class="st">'red'</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span><span class="op">(</span>v <span class="op">=</span> <span class="fl">0</span>, col <span class="op">=</span> <span class="st">'red'</span><span class="op">)</span></code></pre></div>
<p><img src="workshop_introduction_files/figure-html/distance_stat-1.png" width="480" style="display: block; margin: auto;"></p>
</div>
<div id="modeling-the-interaction-of-two-factors-using-independent-perturbations" class="section level2">
<h2 class="hasAnchor">
<a href="#modeling-the-interaction-of-two-factors-using-independent-perturbations" class="anchor"></a>Modeling the interaction of two factors using independent perturbations</h2>
<p>Regulatory Interaction (<span class="math inline">\(RI\)</span>): is the product of the gene statistics from differential expression of the perturbation of the two factors (<span class="math inline">\(x\)</span> and <span class="math inline">\(y\)</span>) separately.</p>
<p><span class="math display">\[
RI_{g} = x_{ge}\times y_{ge}
\]</span> and,</p>
<p><span class="math display">\[
RP_g = \frac{R_{gb}\times RI_{ge}}{n^2}
\]</span></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># define two random stat variables</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">12345</span><span class="op">)</span>; <span class="va">t</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/Normal.html">rnorm</a></span><span class="op">(</span><span class="fl">200</span>, <span class="fl">0</span>, <span class="fl">5</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">321</span><span class="op">)</span>; <span class="va">t2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/Normal.html">rnorm</a></span><span class="op">(</span><span class="fl">200</span>, <span class="fl">0</span>, <span class="fl">5</span><span class="op">)</span>

<span class="co"># plot the two random stat variables</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">t</span>, <span class="va">t2</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">10</span>, <span class="fl">10</span><span class="op">)</span>,
     xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">10</span>, <span class="fl">10</span><span class="op">)</span>, pch <span class="op">=</span> <span class="fl">19</span>, col <span class="op">=</span> <span class="st">'gray'</span>,
     xlab <span class="op">=</span> <span class="st">'rnorm(200, 0, 5)'</span>, ylab <span class="op">=</span> <span class="st">'rnorm(200, 0, 5)'</span><span class="op">)</span>

<span class="co"># highlight four quadrant by sign</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/text.html">text</a></span><span class="op">(</span><span class="fl">5</span>,<span class="fl">5</span>, <span class="st">'(+, +)'</span>, col <span class="op">=</span> <span class="st">'blue'</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/text.html">text</a></span><span class="op">(</span><span class="op">-</span><span class="fl">5</span>,<span class="op">-</span><span class="fl">5</span>, <span class="st">'(-, -)'</span>, col <span class="op">=</span> <span class="st">'blue'</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/text.html">text</a></span><span class="op">(</span><span class="op">-</span><span class="fl">5</span>,<span class="fl">5</span>, <span class="st">'(-, +)'</span>, col <span class="op">=</span> <span class="st">'green'</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/text.html">text</a></span><span class="op">(</span><span class="fl">5</span>,<span class="op">-</span><span class="fl">5</span>, <span class="st">'(+, -)'</span>, col <span class="op">=</span> <span class="st">'green'</span><span class="op">)</span>

<span class="co"># make quadrant lines</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span><span class="op">(</span>h <span class="op">=</span> <span class="fl">0</span>, col <span class="op">=</span> <span class="st">'red'</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span><span class="op">(</span>v <span class="op">=</span> <span class="fl">0</span>, col <span class="op">=</span> <span class="st">'red'</span><span class="op">)</span></code></pre></div>
<p><img src="workshop_introduction_files/figure-html/stat_stat-1.png" width="480" style="display: block; margin: auto;"></p>
</div>
<div id="aggregating-the-effect-of-the-binding-events" class="section level2">
<h2 class="hasAnchor">
<a href="#aggregating-the-effect-of-the-binding-events" class="anchor"></a>Aggregating the effect of the binding events</h2>
<p>Empirical Cumulative Distribution Function (ECDF): the proportion of genes in a category (up- or down-regulated genes) that are ranked at or better than the x-axis (regulatory potential value) value <span class="citation">Tang et al. (2011)</span>.</p>
<p>Aggregate the effect of the factor perturbation in relation to its regulatory potential.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># make a random stat variable</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">12345</span><span class="op">)</span>
<span class="va">t</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/Normal.html">rnorm</a></span><span class="op">(</span><span class="fl">200</span>, <span class="fl">0</span>, <span class="fl">5</span><span class="op">)</span>

<span class="co"># calculate the cumulative distribution function</span>
<span class="va">ecdf_fun</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/ecdf.html">ecdf</a></span><span class="op">(</span><span class="va">t</span><span class="op">)</span>

<span class="co"># plot ecdf</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">ecdf_fun</span>,
     main <span class="op">=</span> <span class="st">''</span>, xlab <span class="op">=</span> <span class="st">'rnorm(200, 0, 5)'</span>, ylab <span class="op">=</span> <span class="st">'ECDF'</span><span class="op">)</span></code></pre></div>
<p><img src="workshop_introduction_files/figure-html/ecdf-1.png" width="480" style="display: block; margin: auto;"></p>
</div>
</div>
<div id="workflow-for-integrating-binding-and-expression-data" class="section level1">
<h1 class="hasAnchor">
<a href="#workflow-for-integrating-binding-and-expression-data" class="anchor"></a>Workflow for integrating binding and expression data</h1>
<div class="figure">
<img src="diagram.jpg" alt=""><p class="caption">Workflow</p>
</div>
</div>
<div id="functions-in-the-target-r-package" class="section level1">
<h1 class="hasAnchor">
<a href="#functions-in-the-target-r-package" class="anchor"></a>Functions in the <code>target</code> <em>R</em> package</h1>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">?</span><span class="fu">target</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/target/man/associated_peaks.html">associated_peaks</a></span>
<span class="op">?</span><span class="fu">target</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/target/man/direct_targets.html">direct_targets</a></span>
<span class="op">?</span><span class="fu">target</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/target/man/plot_predictions.html">plot_predictions</a></span>
<span class="op">?</span><span class="fu">target</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/target/man/test_predictions.html">test_predictions</a></span></code></pre></div>
</div>
<div id="comparison-with-existing-r-packages" class="section level1">
<h1 class="hasAnchor">
<a href="#comparison-with-existing-r-packages" class="anchor"></a>Comparison with existing R packages</h1>
<ul>
<li>
<code>rTRM</code> identifies the transcriptional regulatory modules (TRMs) which are complexes of transcription factors and co-factors by integrating ChIP, gene expression and protein-protein interactions <span class="citation">Diez, Hutchins, and Miranda-Saavedra (2013)</span>.</li>
<li>
<code>TFEA.ChIP</code> curates large quantities of data from different sources and uses this data to build a database where queries of transcription factor targets can be constructed <span class="citation">Puente-Santamaria, Wasserman, and Peso (2019)</span>.</li>
<li>
<code>transcriptR</code> uses the ChIP data to denovo identify transcripts which are then used to quantify the expression from the RNA-Seq data <span class="citation">Karapetyan AR (2019)</span>.</li>
</ul>
</div>
<div id="limitations-of-target" class="section level1">
<h1 class="hasAnchor">
<a href="#limitations-of-target" class="anchor"></a>Limitations of target</h1>
<ul>
<li>Comparable sets of data for the two factors are required; binding data using ChIP and gene expression data under factor perturbation (overexpression or knockdown).</li>
<li>Assume that the interaction between two DNA-binding proteins is linear which may not be the case always.</li>
<li>Cannot detect assisted binding.</li>
</ul>
</div>
<div id="availability" class="section level1">
<h1 class="hasAnchor">
<a href="#availability" class="anchor"></a>Availability</h1>
<ul>
<li>
<code>target</code> is available as an open source R/Bioconductor package, <a href="https://bioconductor.org/packages/target/">here</a>
</li>
<li>An interactive shiny application can be invoked locally through R or accessed directly on the web, <a href="https://mahshaaban.shinyapps.io/target-app/">here</a>
</li>
<li>The source code for the package and the interactive application is available <a href="https://github.com/MahShaaban/target">here</a>
</li>
</ul>
<div id="code-walkthrough-next" class="section level3">
<h3 class="hasAnchor">
<a href="#code-walkthrough-next" class="anchor"></a>Code Walkthrough (–&gt; Next)</h3>
</div>
<div id="references" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Breitling2004" class="csl-entry">
Breitling, Rainer, Patrick Armengaud, Anna Amtmann, and Pawel Herzyk. 2004. <span>“<span class="nocase">Rank products: a simple, yet powerful, new method to detect differentially regulated genes in replicated microarray experiments</span>.”</span> <em>FEBS Letters</em> 573 (1-3): 83–92. <a href="https://doi.org/10.1016/j.febslet.2004.07.055">https://doi.org/10.1016/j.febslet.2004.07.055</a>.
</div>
<div id="ref-Diez2013" class="csl-entry">
Diez, Diego, Andrew Paul Hutchins, and Diego Miranda-Saavedra. 2013. <span>“<span class="nocase">Systematic identification of transcriptional regulatory modules from protein-protein interaction networks</span>.”</span> <em>Nucleic Acids Research</em>. <a href="https://doi.org/10.1093/nar/gkt913">https://doi.org/10.1093/nar/gkt913</a>.
</div>
<div id="ref-KarapetyanAR2019" class="csl-entry">
Karapetyan AR. 2019. <span>“<span class="nocase">An Integrative Tool for ChIP- And RNA-Seq Based Primary Transcripts Detection and Quantification</span>.”</span> R package. <a href="https://doi.org/10.18129/B9.bioc.transcriptR">https://doi.org/10.18129/B9.bioc.transcriptR</a>.
</div>
<div id="ref-Puente-Santamaria2019" class="csl-entry">
Puente-Santamaria, Laura, Wyeth W Wasserman, and Luis del Peso. 2019. <span>“<span class="nocase">TFEA.ChIP: a tool kit for transcription factor binding site enrichment analysis capitalizing on ChIP-seq datasets</span>.”</span> <em>Bioinformatics</em>. <a href="https://doi.org/10.1093/bioinformatics/btz573">https://doi.org/10.1093/bioinformatics/btz573</a>.
</div>
<div id="ref-Rougemont2012" class="csl-entry">
Rougemont, Jacques, and Felix Naef. 2012. <span>“<span class="nocase">Computational Analysis of Protein–DNA Interactions from ChIP-seq Data</span>.”</span> In <em>Gene Regulatory Networks</em>, 263–73. Springer. <a href="https://doi.org/10.1007/978-1-61779-292-2_16">https://doi.org/10.1007/978-1-61779-292-2_16</a>.
</div>
<div id="ref-Tang2011" class="csl-entry">
Tang, Qianzi, Yiwen Chen, Clifford Meyer, Tim Geistlinger, Mathieu Lupien, Qian Wang, Tao Liu, Yong Zhang, Myles Brown, and Xiaole Shirley Liu. 2011. <span>“<span class="nocase">A comprehensive view of nuclear receptor cancer cistromes.</span>”</span> <em>Cancer Research</em> 71 (22): 6940–47. <a href="https://doi.org/10.1158/0008-5472.CAN-11-2091">https://doi.org/10.1158/0008-5472.CAN-11-2091</a>.
</div>
<div id="ref-Wang2013b" class="csl-entry">
Wang, Su, Hanfei Sun, Jian Ma, Chongzhi Zang, Chenfei Wang, Juan Wang, Qianzi Tang, Clifford A. Meyer, Yong Zhang, and X. Shirley Liu. 2013. <span>“<span class="nocase">Target analysis by integration of transcriptome and ChIP-seq data with BETA</span>.”</span> <em>Nature Protocols</em>. <a href="https://doi.org/10.1038/nprot.2013.150">https://doi.org/10.1038/nprot.2013.150</a>.
</div>
</div>
</div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p><a href="mailto:mahshaaban@gnu.ac.kr" class="email">mahshaaban@gnu.ac.kr</a><a href="#fnref1" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Mahmoud Ahmed.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
