<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction: Bioconductor packages and classes • IntegratingDataBioc</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction: Bioconductor packages and classes">
<meta property="og:description" content="IntegratingDataBioc">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-93043521-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-93043521-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">IntegratingDataBioc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/workshop_code.html">Code Walkthrough: A use case of YY1 and YY2 in HeLa cells</a>
    </li>
    <li>
      <a href="../articles/workshop_introduction.html">Introduction: Introduction to target package</a>
    </li>
    <li>
      <a href="../articles/workshop_packages.html">Introduction: Bioconductor packages and classes</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/MahShaaban/IntegratingDataBioc">
    <span class="fas fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="workshop_packages_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction: Bioconductor packages and classes</h1>
                        <h4 class="author">Mahmoud Ahmed<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>
</h4>
            
            <h4 class="date">2021-02-27</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/MahShaaban/IntegratingDataBioc/blob/master/vignettes/workshop_packages.Rmd"><code>vignettes/workshop_packages.Rmd</code></a></small>
      <div class="hidden name"><code>workshop_packages.Rmd</code></div>

    </div>

    
    
<div id="overview" class="section level1">
<h1 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h1>
<p>This is a brief introduction to Bioconductor. We will cover the basics of the Bioconductor <strong>packages</strong> and <strong>classes</strong> that are relevant to the workshop.</p>
</div>
<div id="types-of-packages" class="section level1">
<h1 class="hasAnchor">
<a href="#types-of-packages" class="anchor"></a>Types of Packages</h1>
<ul>
<li>Software</li>
<li>Annotation</li>
<li>Experiment data</li>
</ul>
<div id="software-packages" class="section level2">
<h2 class="hasAnchor">
<a href="#software-packages" class="anchor"></a>Software Packages</h2>
<p>Provide implementation of algorithms (e.g. statistical analysis), access to resources (e.g. biomart, or NCBI) or visualizations (e.g. volcano plots, pathways plots).</p>
</div>
<div id="annotation-packages" class="section level2">
<h2 class="hasAnchor">
<a href="#annotation-packages" class="anchor"></a>Annotation Packages</h2>
<p>Database-like packages that provide information linking identifiers (e.g., Entrez gene names or Affymetrix probe ids) to other information (e.g., chromosomal location, Gene Ontology category). It is also encouraged to utilize AnnotationHub for storage and access to large raw data files and their conversion to standard R formats.</p>
</div>
<div id="experiment-data-packages" class="section level2">
<h2 class="hasAnchor">
<a href="#experiment-data-packages" class="anchor"></a>Experiment Data Packages</h2>
<p>provide data sets that are used, often by software packages, to illustrate particular analyses. These packages contain curated data from an experiment, teaching course or publication and in most cases contain a single data set. It is also encouraged to utilize ExperimentHub for storage and access to larger data files.</p>
</div>
</div>
<div id="bioconductor-classes" class="section level1">
<h1 class="hasAnchor">
<a href="#bioconductor-classes" class="anchor"></a>Bioconductor Classes</h1>
<ul>
<li>Three-table structures</li>
</ul>
<div class="figure">
<img src="three_tables.svg" alt=""><p class="caption">Summarized Experiment Object</p>
</div>
<ul>
<li>Accessors and constructors</li>
</ul>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># constructors</span>
<span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu">SomeObject</span><span class="op">(</span><span class="va">data</span>, <span class="va">metadata</span><span class="op">)</span>

<span class="co"># accessors</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></code></pre></div>
</div>
<div id="packages-relevant-to-the-workshop" class="section level1">
<h1 class="hasAnchor">
<a href="#packages-relevant-to-the-workshop" class="anchor"></a>Packages relevant to the workshop</h1>
<div id="genomicranges" class="section level2">
<h2 class="hasAnchor">
<a href="#genomicranges" class="anchor"></a><code>GenomicRanges</code>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># load library</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicRanges">GenomicRanges</a></span><span class="op">)</span>

<span class="co"># build an IRanges object</span>
<span class="va">regions</span> <span class="op">&lt;-</span> <span class="fu">IRanges</span><span class="op">(</span>start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">10</span><span class="op">)</span>,
                   end   <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">11</span>, <span class="fl">20</span><span class="op">)</span>,
                   names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'region_1'</span>, <span class="st">'region_2'</span><span class="op">)</span><span class="op">)</span>

<span class="va">regions</span>
<span class="co">#&gt; IRanges object with 2 ranges and 0 metadata columns:</span>
<span class="co">#&gt;                start       end     width</span>
<span class="co">#&gt;            &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;</span>
<span class="co">#&gt;   region_1         1        11        11</span>
<span class="co">#&gt;   region_2        10        20        11</span>

<span class="co"># build a GRanges object</span>
<span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'chr1'</span>, <span class="st">'chr1'</span><span class="op">)</span>,
        ranges <span class="op">=</span> <span class="va">regions</span><span class="op">)</span>
<span class="co">#&gt; GRanges object with 2 ranges and 0 metadata columns:</span>
<span class="co">#&gt;            seqnames    ranges strand</span>
<span class="co">#&gt;               &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;   region_1     chr1      1-11      *</span>
<span class="co">#&gt;   region_2     chr1     10-20      *</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>

<span class="co"># add strand info to the object</span>
<span class="va">gr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'chr1'</span>, <span class="st">'chr1'</span><span class="op">)</span>,
              ranges <span class="op">=</span> <span class="va">regions</span>,
              strand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'+'</span>, <span class="st">'+'</span><span class="op">)</span><span class="op">)</span>

<span class="va">gr</span>
<span class="co">#&gt; GRanges object with 2 ranges and 0 metadata columns:</span>
<span class="co">#&gt;            seqnames    ranges strand</span>
<span class="co">#&gt;               &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;   region_1     chr1      1-11      +</span>
<span class="co">#&gt;   region_2     chr1     10-20      +</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>

<span class="co"># add metadata to the object</span>
<span class="va">gr</span><span class="op">$</span><span class="va">region_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'first'</span>, <span class="st">'second'</span><span class="op">)</span>

<span class="va">gr</span>
<span class="co">#&gt; GRanges object with 2 ranges and 1 metadata column:</span>
<span class="co">#&gt;            seqnames    ranges strand | region_name</span>
<span class="co">#&gt;               &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span>
<span class="co">#&gt;   region_1     chr1      1-11      + |       first</span>
<span class="co">#&gt;   region_2     chr1     10-20      + |      second</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>

<span class="co"># accessors</span>
<span class="fu">seqnames</span><span class="op">(</span><span class="va">gr</span><span class="op">)</span>
<span class="co">#&gt; factor-Rle of length 2 with 1 run</span>
<span class="co">#&gt;   Lengths:    2</span>
<span class="co">#&gt;   Values : chr1</span>
<span class="co">#&gt; Levels(1): chr1</span>
<span class="fu"><a href="https://rdrr.io/r/stats/start.html">start</a></span><span class="op">(</span><span class="va">gr</span><span class="op">)</span>
<span class="co">#&gt; [1]  1 10</span>
<span class="fu"><a href="https://rdrr.io/r/stats/start.html">end</a></span><span class="op">(</span><span class="va">gr</span><span class="op">)</span>
<span class="co">#&gt; [1] 11 20</span>
<span class="fu">strand</span><span class="op">(</span><span class="va">gr</span><span class="op">)</span>
<span class="co">#&gt; factor-Rle of length 2 with 1 run</span>
<span class="co">#&gt;   Lengths: 2</span>
<span class="co">#&gt;   Values : +</span>
<span class="co">#&gt; Levels(3): + - *</span>
<span class="fu">width</span><span class="op">(</span><span class="va">gr</span><span class="op">)</span>
<span class="co">#&gt; [1] 11 11</span>
<span class="fu">mcols</span><span class="op">(</span><span class="va">gr</span><span class="op">)</span>
<span class="co">#&gt; DataFrame with 2 rows and 1 column</span>
<span class="co">#&gt;          region_name</span>
<span class="co">#&gt;          &lt;character&gt;</span>
<span class="co">#&gt; region_1       first</span>
<span class="co">#&gt; region_2      second</span></code></pre></div>
</div>
<div id="summarizedexperiments" class="section level2">
<h2 class="hasAnchor">
<a href="#summarizedexperiments" class="anchor"></a><code>SummarizedExperiments</code>
</h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># load library</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment">SummarizedExperiment</a></span><span class="op">)</span>

<span class="co"># make a data.frame with phenotype data</span>
<span class="va">pd</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">'sample'</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, sep <span class="op">=</span> <span class="st">'_'</span><span class="op">)</span>,
                 group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'control'</span>, <span class="st">'treatment'</span><span class="op">)</span>, each <span class="op">=</span> <span class="fl">2</span><span class="op">)</span><span class="op">)</span>

<span class="co"># make a 2 by 4 matrix with values</span>
<span class="va">mat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Normal.html">rnorm</a></span><span class="op">(</span><span class="fl">8</span><span class="op">)</span>, nrow <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>

<span class="co"># build a SummarizedExperiment object</span>
<span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html">SummarizedExperiment</a></span><span class="op">(</span>assays <span class="op">=</span> <span class="va">mat</span>,
                           rowData <span class="op">=</span> <span class="va">gr</span>,
                           colData <span class="op">=</span> <span class="va">pd</span><span class="op">)</span>
<span class="va">se</span>
<span class="co">#&gt; class: RangedSummarizedExperiment </span>
<span class="co">#&gt; dim: 2 4 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(1): ''</span>
<span class="co">#&gt; rownames(2): region_1 region_2</span>
<span class="co">#&gt; rowData names(1): region_name</span>
<span class="co">#&gt; colnames: NULL</span>
<span class="co">#&gt; colData names(2): id group</span>

<span class="co"># accessors</span>
<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assay</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span>
<span class="co">#&gt;                [,1]        [,2]       [,3]       [,4]</span>
<span class="co">#&gt; region_1  0.4703326  0.03897944 -0.4338181  0.2364776</span>
<span class="co">#&gt; region_2 -0.1196687 -0.63810615 -0.0357938 -0.2908488</span>
<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">rowData</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span>
<span class="co">#&gt; DataFrame with 2 rows and 1 column</span>
<span class="co">#&gt;          region_name</span>
<span class="co">#&gt;          &lt;character&gt;</span>
<span class="co">#&gt; region_1       first</span>
<span class="co">#&gt; region_2      second</span>
<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html">rowRanges</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span>
<span class="co">#&gt; GRanges object with 2 ranges and 1 metadata column:</span>
<span class="co">#&gt;            seqnames    ranges strand | region_name</span>
<span class="co">#&gt;               &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span>
<span class="co">#&gt;   region_1     chr1      1-11      + |       first</span>
<span class="co">#&gt;   region_2     chr1     10-20      + |      second</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span>
<span class="co">#&gt; DataFrame with 4 rows and 2 columns</span>
<span class="co">#&gt;            id       group</span>
<span class="co">#&gt;   &lt;character&gt; &lt;character&gt;</span>
<span class="co">#&gt; 1    sample_1     control</span>
<span class="co">#&gt; 2    sample_2     control</span>
<span class="co">#&gt; 3    sample_1   treatment</span>
<span class="co">#&gt; 4    sample_2   treatment</span></code></pre></div>
</div>
<div id="rtracklayer" class="section level2">
<h2 class="hasAnchor">
<a href="#rtracklayer" class="anchor"></a><code>rtracklayer</code>
</h2>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># load library</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">rtracklayer</span><span class="op">)</span>

<span class="co"># export a GRanges object as a bed file</span>
<span class="fu"><a href="https://rdrr.io/pkg/rtracklayer/man/BEDFile-class.html">export.bed</a></span><span class="op">(</span><span class="va">gr</span>, <span class="st">'regions.bed'</span><span class="op">)</span>

<span class="co"># import a bed file as GRanges</span>
<span class="fu"><a href="https://rdrr.io/pkg/rtracklayer/man/BEDFile-class.html">import.bed</a></span><span class="op">(</span><span class="st">'regions.bed'</span><span class="op">)</span>
<span class="co">#&gt; GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |        name     score</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr1      1-11      + |    region_1         0</span>
<span class="co">#&gt;   [2]     chr1     10-20      + |    region_2         0</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></code></pre></div>
</div>
<div id="org-hs-eg-db" class="section level2">
<h2 class="hasAnchor">
<a href="#org-hs-eg-db" class="anchor"></a><code>org.Hs.eg.db</code>
</h2>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># load library</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span>

<span class="co"># shorter name</span>
<span class="va">org</span> <span class="op">&lt;-</span> <span class="va">org.Hs.eg.db</span>
<span class="va">org</span>
<span class="co">#&gt; OrgDb object:</span>
<span class="co">#&gt; | DBSCHEMAVERSION: 2.1</span>
<span class="co">#&gt; | Db type: OrgDb</span>
<span class="co">#&gt; | Supporting package: AnnotationDbi</span>
<span class="co">#&gt; | DBSCHEMA: HUMAN_DB</span>
<span class="co">#&gt; | ORGANISM: Homo sapiens</span>
<span class="co">#&gt; | SPECIES: Human</span>
<span class="co">#&gt; | EGSOURCEDATE: 2020-Sep23</span>
<span class="co">#&gt; | EGSOURCENAME: Entrez Gene</span>
<span class="co">#&gt; | EGSOURCEURL: ftp://ftp.ncbi.nlm.nih.gov/gene/DATA</span>
<span class="co">#&gt; | CENTRALID: EG</span>
<span class="co">#&gt; | TAXID: 9606</span>
<span class="co">#&gt; | GOSOURCENAME: Gene Ontology</span>
<span class="co">#&gt; | GOSOURCEURL: http://current.geneontology.org/ontology/go-basic.obo</span>
<span class="co">#&gt; | GOSOURCEDATE: 2020-09-10</span>
<span class="co">#&gt; | GOEGSOURCEDATE: 2020-Sep23</span>
<span class="co">#&gt; | GOEGSOURCENAME: Entrez Gene</span>
<span class="co">#&gt; | GOEGSOURCEURL: ftp://ftp.ncbi.nlm.nih.gov/gene/DATA</span>
<span class="co">#&gt; | KEGGSOURCENAME: KEGG GENOME</span>
<span class="co">#&gt; | KEGGSOURCEURL: ftp://ftp.genome.jp/pub/kegg/genomes</span>
<span class="co">#&gt; | KEGGSOURCEDATE: 2011-Mar15</span>
<span class="co">#&gt; | GPSOURCENAME: UCSC Genome Bioinformatics (Homo sapiens)</span>
<span class="co">#&gt; | GPSOURCEURL: </span>
<span class="co">#&gt; | GPSOURCEDATE: 2020-Aug27</span>
<span class="co">#&gt; | ENSOURCEDATE: 2020-Aug18</span>
<span class="co">#&gt; | ENSOURCENAME: Ensembl</span>
<span class="co">#&gt; | ENSOURCEURL: ftp://ftp.ensembl.org/pub/current_fasta</span>
<span class="co">#&gt; | UPSOURCENAME: Uniprot</span>
<span class="co">#&gt; | UPSOURCEURL: http://www.UniProt.org/</span>
<span class="co">#&gt; | UPSOURCEDATE: Mon Oct  5 00:18:02 2020</span>

<span class="co"># show names of columns</span>
<span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html">columns</a></span><span class="op">(</span><span class="va">org</span><span class="op">)</span>
<span class="co">#&gt;  [1] "ACCNUM"       "ALIAS"        "ENSEMBL"      "ENSEMBLPROT"  "ENSEMBLTRANS"</span>
<span class="co">#&gt;  [6] "ENTREZID"     "ENZYME"       "EVIDENCE"     "EVIDENCEALL"  "GENENAME"    </span>
<span class="co">#&gt; [11] "GO"           "GOALL"        "IPI"          "MAP"          "OMIM"        </span>
<span class="co">#&gt; [16] "ONTOLOGY"     "ONTOLOGYALL"  "PATH"         "PFAM"         "PMID"        </span>
<span class="co">#&gt; [21] "PROSITE"      "REFSEQ"       "SYMBOL"       "UCSCKG"       "UNIGENE"     </span>
<span class="co">#&gt; [26] "UNIPROT"</span></code></pre></div>
</div>
<div id="txdb-hsapiens-ucsc-hg19-knowngene" class="section level2">
<h2 class="hasAnchor">
<a href="#txdb-hsapiens-ucsc-hg19-knowngene" class="anchor"></a><code>TxDb.Hsapiens.UCSC.hg19.knownGene</code>
</h2>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># load library</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">TxDb.Hsapiens.UCSC.hg19.knownGene</span><span class="op">)</span>

<span class="co"># shorter name</span>
<span class="va">txdb</span> <span class="op">&lt;-</span> <span class="va">TxDb.Hsapiens.UCSC.hg19.knownGene</span>
<span class="va">txdb</span>
<span class="co">#&gt; TxDb object:</span>
<span class="co">#&gt; # Db type: TxDb</span>
<span class="co">#&gt; # Supporting package: GenomicFeatures</span>
<span class="co">#&gt; # Data source: UCSC</span>
<span class="co">#&gt; # Genome: hg19</span>
<span class="co">#&gt; # Organism: Homo sapiens</span>
<span class="co">#&gt; # Taxonomy ID: 9606</span>
<span class="co">#&gt; # UCSC Table: knownGene</span>
<span class="co">#&gt; # Resource URL: http://genome.ucsc.edu/</span>
<span class="co">#&gt; # Type of Gene ID: Entrez Gene ID</span>
<span class="co">#&gt; # Full dataset: yes</span>
<span class="co">#&gt; # miRBase build ID: GRCh37</span>
<span class="co">#&gt; # transcript_nrow: 82960</span>
<span class="co">#&gt; # exon_nrow: 289969</span>
<span class="co">#&gt; # cds_nrow: 237533</span>
<span class="co">#&gt; # Db created by: GenomicFeatures package from Bioconductor</span>
<span class="co">#&gt; # Creation time: 2015-10-07 18:11:28 +0000 (Wed, 07 Oct 2015)</span>
<span class="co">#&gt; # GenomicFeatures version at creation time: 1.21.30</span>
<span class="co">#&gt; # RSQLite version at creation time: 1.0.0</span>
<span class="co">#&gt; # DBSCHEMAVERSION: 1.1</span>

<span class="co"># show names of columns</span>
<span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html">columns</a></span><span class="op">(</span><span class="va">txdb</span><span class="op">)</span>
<span class="co">#&gt;  [1] "CDSCHROM"   "CDSEND"     "CDSID"      "CDSNAME"    "CDSSTART"  </span>
<span class="co">#&gt;  [6] "CDSSTRAND"  "EXONCHROM"  "EXONEND"    "EXONID"     "EXONNAME"  </span>
<span class="co">#&gt; [11] "EXONRANK"   "EXONSTART"  "EXONSTRAND" "GENEID"     "TXCHROM"   </span>
<span class="co">#&gt; [16] "TXEND"      "TXID"       "TXNAME"     "TXSTART"    "TXSTRAND"  </span>
<span class="co">#&gt; [21] "TXTYPE"</span></code></pre></div>
</div>
<div id="annotationdbi" class="section level2">
<h2 class="hasAnchor">
<a href="#annotationdbi" class="anchor"></a><code>AnnotationDbi</code>
</h2>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># load library</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/AnnotationDbi">AnnotationDbi</a></span><span class="op">)</span>

<span class="co"># map gene symbols to entrez ids from the organism pkg</span>
<span class="va">gene_ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html">select</a></span><span class="op">(</span><span class="va">org</span>,
                   keys <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'ATG5'</span>, <span class="st">'ATG7'</span><span class="op">)</span>,
                   keytype <span class="op">=</span> <span class="st">'SYMBOL'</span>,
                   columns <span class="op">=</span> <span class="st">'ENTREZID'</span><span class="op">)</span>
<span class="va">gene_ids</span>
<span class="co">#&gt;   SYMBOL ENTREZID</span>
<span class="co">#&gt; 1   ATG5     9474</span>
<span class="co">#&gt; 2   ATG7    10533</span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># load library</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicFeatures">GenomicFeatures</a></span><span class="op">)</span>

<span class="co"># extract gene coordinates from the taxonomy pkg</span>
<span class="va">gene_coordinates</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html">genes</a></span><span class="op">(</span><span class="va">txdb</span>,
                          filter <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>gene_id <span class="op">=</span> <span class="va">gene_ids</span><span class="op">$</span><span class="va">ENTREZID</span><span class="op">)</span><span class="op">)</span>

<span class="va">gene_coordinates</span>
<span class="co">#&gt; GRanges object with 2 ranges and 1 metadata column:</span>
<span class="co">#&gt;         seqnames              ranges strand |     gene_id</span>
<span class="co">#&gt;            &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span>
<span class="co">#&gt;   10533     chr3   11314010-11599139      + |       10533</span>
<span class="co">#&gt;    9474     chr6 106632352-106773695      - |        9474</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 93 sequences (1 circular) from hg19 genome</span>

<span class="co"># extract promoter regions</span>
<span class="va">gene_promoters</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html">promoters</a></span><span class="op">(</span><span class="va">gene_coordinates</span>,
                            upstream <span class="op">=</span> <span class="fl">3000</span><span class="op">)</span>

<span class="va">gene_promoters</span>
<span class="co">#&gt; GRanges object with 2 ranges and 1 metadata column:</span>
<span class="co">#&gt;         seqnames              ranges strand |     gene_id</span>
<span class="co">#&gt;            &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span>
<span class="co">#&gt;   10533     chr3   11311010-11314209      + |       10533</span>
<span class="co">#&gt;    9474     chr6 106773496-106776695      - |        9474</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 93 sequences (1 circular) from hg19 genome</span></code></pre></div>
</div>
<div id="target" class="section level2">
<h2 class="hasAnchor">
<a href="#target" class="anchor"></a><code>target</code>
</h2>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># load library</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/MahShaaban/target">target</a></span><span class="op">)</span>

<span class="co"># get help pages</span>
<span class="op">?</span><span class="va">target</span>
<span class="op">?</span><span class="fu">target</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/target/man/associated_peaks.html">associated_peaks</a></span>

<span class="co"># explore package vignettes</span>
<span class="fu"><a href="https://rdrr.io/r/utils/vignette.html">vignette</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">'target'</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/vignette.html">vignette</a></span><span class="op">(</span><span class="st">'target'</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/vignette.html">vignette</a></span><span class="op">(</span><span class="st">'extend-target'</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p><a href="mailto:mahshaaban@gnu.ac.kr" class="email">mahshaaban@gnu.ac.kr</a><a href="#fnref1" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Mahmoud Ahmed.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
